# Think Python

## Chapter 9 Case study: word play

HTML version can be found [here](http://greenteapress.com/thinkpython2/html/thinkpython2010.html "Chpt 9").

### 9.1 Reading word lists

*Importing the word list from [this link](http://thinkpython2.com/code/words.txt "words.txt"), and using `open()` to open the file:*

In [1]:
import os
path = "C:\\Users\\mjcor\\Desktop\\ProgrammingStuff\\ThinkPython"
os.chdir(path)
fin = open('words.txt')

*`.readline()` reads characters until it gets to a newline, and then returns the result as a string:*

In [2]:
fin.readline()

'aa\n'

*`.readline()` will remember our place in the file:*


In [3]:
fin.readline()

'aah\n'

*Using `.strip()` to get rid of the newline character:*

In [4]:
fin.readline().strip()

'aahed'

*Here's a `for`-loop that will read each word in `words.txt` and print it, one word per line:*

```
fin = open('words.txt')
for line in fin:
    print(line.strip())
```

*It takes quite a while for the iPython interpreter at GitHub to iterate through all the words in the list, so for the sake of speed, the next cell uses a `for` loop to go through just the first 100.*

In [5]:
fin = open('words.txt')
i = 0
for line in fin:   
    if i < 100:
        print(line.strip())
    i += 1


aa
aah
aahed
aahing
aahs
aal
aalii
aaliis
aals
aardvark
aardvarks
aardwolf
aardwolves
aas
aasvogel
aasvogels
aba
abaca
abacas
abaci
aback
abacus
abacuses
abaft
abaka
abakas
abalone
abalones
abamp
abampere
abamperes
abamps
abandon
abandoned
abandoning
abandonment
abandonments
abandons
abas
abase
abased
abasedly
abasement
abasements
abaser
abasers
abases
abash
abashed
abashes
abashing
abasing
abatable
abate
abated
abatement
abatements
abater
abaters
abates
abating
abatis
abatises
abator
abators
abattis
abattises
abattoir
abattoirs
abaxial
abaxile
abbacies
abbacy
abbatial
abbe
abbes
abbess
abbesses
abbey
abbeys
abbot
abbotcies
abbotcy
abbots
abbreviate
abbreviated
abbreviates
abbreviating
abbreviation
abbreviations
abdicate
abdicated
abdicates
abdicating
abdication
abdications
abdomen
abdomens
abdomina
abdominal


### 9.2  Exercises
*There are solutions to these exercises in the next section. You should at least attempt each one before you read the solutions.*

#### Exercise 1   

*Write a program that reads `words.txt` and prints only the words with more than 20 characters (not counting whitespace).*

In [6]:
fin = open('words.txt')
for line in fin:
    word = line.strip()
    if len(word) > 20:
        print(word)

counterdemonstrations
hyperaggressivenesses
microminiaturizations


#### Exercise 2  

*In 1939 Ernest Vincent Wright published a 50,000 word novel called Gadsby that does not contain the letter “e”. Since “e” is the most common letter in English, that’s not easy to do.*

*In fact, it is difficult to construct a solitary thought without using that most common symbol. It is slow going at first, but with caution and hours of training you can gradually gain facility.*

*All right, I’ll stop now.*

*Write a function called `has_no_e` that returns `True` if the given word doesn’t have the letter “e” in it.*

*Write a program that reads `words.txt` and prints only the words that have no “e”. Compute the percentage of words in the list that have no “e”.*


In [7]:
def has_no_e(word):
    """
    Returns False if word contains the character 'e'.
    """
    
    if "e" in word:
        return False
    return True

In [8]:
has_no_e("banana")

True

In [9]:
has_no_e("grape")

False

In [10]:
fin = open('words.txt')

# Tallying and printing words in words.txt without 'e'

no_e_tally = 0

for line in fin:
    word = line.strip()
    if has_no_e(word):
        print(word)
        no_e_tally += 1
        
no_e_tally

aa
aah
aahing
aahs
aal
aalii
aaliis
aals
aardvark
aardvarks
aardwolf
aas
aba
abaca
abacas
abaci
aback
abacus
abaft
abaka
abakas
abamp
abamps
abandon
abandoning
abandons
abas
abash
abashing
abasing
abating
abatis
abator
abators
abattis
abattoir
abattoirs
abaxial
abbacy
abbatial
abbot
abbotcy
abbots
abdicating
abdication
abdications
abdomina
abdominal
abdominally
abducing
abduct
abducting
abductor
abductors
abducts
abfarad
abfarads
abhor
abhorring
abhors
abiding
abigail
abigails
ability
abiosis
abiotic
abjuration
abjurations
abjuring
ablating
ablation
ablations
ablaut
ablauts
ablings
ablins
abloom
ablush
ablution
ablutions
ably
abmho
abmhos
abnormal
abnormality
abnormally
abnormals
abo
aboard
aboding
abohm
abohms
aboil
abolish
abolishing
abolition
abolitions
abolla
aboma
abomas
abomasa
abomasal
abomasi
abomasum
abomasus
abominating
abomination
abominations
aboon
aboral
aborally
aboriginal
aborning
abort
aborting
abortion
abortions
aborts
abos
abought
aboulia
aboulias
aboulic
abound
aboun

applicators
apply
applying
appoint
appointing
appoints
apportion
apportioning
apportions
apposing
appraisal
appraisals
appraising
apprising
apprizing
approach
approaching
approbation
appropriating
appropriation
appropriations
approval
approvals
approving
approximating
approximation
approximations
apractic
apraxia
apraxias
apraxic
apricot
apricots
apron
aproning
aprons
apropos
apsidal
apsis
apt
aptly
aqua
aquanaut
aquanauts
aquaria
aquarial
aquarian
aquarians
aquarist
aquarists
aquarium
aquariums
aquas
aquatic
aquatics
aquatint
aquatinting
aquatints
aquavit
aquavits
ar
arabizing
arachnid
arachnids
arak
araks
arapaima
arapaimas
araroba
ararobas
arbalist
arbalists
arbitral
arbitrarily
arbitrary
arbitrating
arbitration
arbitrations
arbitrator
arbitrators
arbor
arborist
arborists
arborizing
arborous
arbors
arbour
arbours
arbutus
arc
arcadia
arcadian
arcadians
arcadias
arcading
arcadings
arcana
arcanum
arch
archaic
archaically
archaising
archaism
archaisms
archaist
archaists
archaizing
archb

bazaar
bazaars
bazar
bazars
bazooka
bazookas
bhakta
bhaktas
bhakti
bhaktis
bhang
bhangs
bhoot
bhoots
bhut
bhuts
bi
bialy
bialys
biannual
biannually
bias
biasing
biassing
biathlon
biathlons
biaxal
biaxial
bib
bibasic
bibasilar
bibb
bibbing
bibbs
bibcock
bibcocks
biblical
bibliographic
bibliographical
bibliography
bibs
bibulous
bicarb
bicarbs
bicolor
bicolors
bicolour
bicolours
biconcavity
bicorn
bicron
bicrons
bicultural
bicuspid
bicuspids
bicyclic
bicycling
bid
bidarka
bidarkas
biddably
bidding
biddings
biddy
biding
bids
bifacial
biff
biffin
biffing
biffins
biffs
biffy
bifid
bifidity
bifidly
bifilar
bifocal
bifocals
bifold
biform
bifunctional
big
bigamist
bigamists
bigamous
bigamy
bigaroon
bigaroons
biggin
bigging
biggings
biggins
biggish
biggity
bighorn
bighorns
bight
bighting
bights
bigly
bigmouth
bigmouths
bignonia
bignonias
bigot
bigotry
bigots
bigwig
bigwigs
bihourly
bijou
bijous
bijoux
bijugous
biking
bikini
bikinis
bilabial
bilabials
bilbo
bilboa
bilboas
bilbos
bilging
bilgy
bil

bumpkins
bumps
bumpy
bums
bun
bunch
bunchily
bunching
bunchy
bunco
buncoing
buncos
bund
bundist
bundists
bundling
bundlings
bunds
bung
bungalow
bungalows
bunging
bungling
bunglings
bungs
bunion
bunions
bunk
bunking
bunko
bunkoing
bunkos
bunks
bunkum
bunkums
bunky
bunn
bunns
bunny
buns
bunt
bunting
buntings
bunts
bunya
bunyas
buoy
buoyancy
buoyant
buoying
buoys
buqsha
buqshas
bur
bura
buran
burans
buras
burbling
burbly
burbot
burbots
burd
burdock
burdocks
burds
burg
burgh
burghal
burghs
burglar
burglarizing
burglars
burglary
burgling
burgoo
burgoos
burgout
burgouts
burgs
burgundy
burial
burials
burin
burins
burking
burl
burlap
burlaps
burlily
burling
burls
burly
burn
burning
burnings
burnish
burnishing
burnous
burnout
burnouts
burns
burnt
burp
burping
burps
burr
burring
burro
burros
burrow
burrowing
burrows
burrs
burry
burs
bursa
bursal
bursar
bursars
bursary
bursas
bursitis
burst
bursting
bursts
burton
burtons
bury
burying
bus
busboy
busboys
busby
bush
bushbuck
bushbucks
bushgoat
bushg

chins
chints
chintz
chintzy
chip
chipmuck
chipmucks
chipmunk
chipmunks
chipping
chippy
chips
chirk
chirking
chirks
chirm
chirming
chirms
chiro
chiropodist
chiropodists
chiropody
chiropractic
chiropractics
chiropractor
chiropractors
chiros
chirp
chirpily
chirping
chirps
chirpy
chirr
chirring
chirrs
chirrup
chirruping
chirrups
chirrupy
chis
chit
chital
chitchat
chitchats
chitchatting
chitin
chitins
chitlin
chitling
chitlings
chitlins
chiton
chitons
chits
chitty
chivalric
chivalrous
chivalrously
chivalry
chivari
chivariing
chivaris
chivvy
chivvying
chivy
chivying
chlamys
chloral
chlorals
chlorambucil
chlordan
chlordans
chloric
chlorid
chlorids
chlorin
chlorinating
chlorination
chlorinations
chlorinator
chlorinators
chlorins
chloroform
chloroforming
chloroforms
chlorophyll
chlorophylls
chlorous
chock
chockfull
chocking
chocks
choir
choirboy
choirboys
choiring
choirs
choking
choky
cholla
chollas
chomp
chomping
chomps
chon
choosing
choosy
chop
chopin
chopins
choppily
chopping
choppy
chops
ch

craals
crab
crabbing
crabby
crabs
crack
crackdown
crackdowns
cracking
crackings
crackling
crackly
crackpot
crackpots
cracks
crackup
crackups
cracky
cradling
craft
craftily
crafting
crafts
craftsman
craftsmanship
craftsmanships
crafty
crag
craggily
craggy
crags
cragsman
cram
crambo
crambos
cramming
cramoisy
cramp
cramping
crampit
crampits
crampon
crampons
crampoon
crampoons
cramps
crams
cranch
cranching
crania
cranial
craning
cranium
craniums
crank
crankily
cranking
crankling
crankly
crankous
crankpin
crankpins
cranks
cranky
crannog
crannogs
cranny
crap
craping
crapping
crappy
craps
crash
crashing
crasis
crass
crassly
cratch
crating
craton
cratonic
cratons
craunch
craunching
cravat
cravats
craving
cravings
craw
crawdad
crawdads
crawfish
crawfishing
crawl
crawling
crawls
crawlway
crawlways
crawly
craws
crayfish
crayon
crayoning
crayons
crazily
crazing
crazy
crib
cribbing
cribbings
cribrous
cribs
cribwork
cribworks
crick
cricking
cricks
cricoid
cricoids
criminal
criminals
crimp
crimping
c

disdainful
disdainfully
disdaining
disdains
disfavor
disfavoring
disfavors
disfiguring
disfranchising
disfrock
disfrocking
disfrocks
disgorging
disgracing
disguising
disgust
disgusting
disgustingly
disgusts
dish
disharmonious
disharmony
dishcloth
dishcloths
dishful
dishfuls
dishing
dishonor
dishonorably
dishonoring
dishonors
dishpan
dishpans
dishrag
dishrags
dishy
disillusion
disillusioning
disillusions
disinclination
disinclinations
disinclining
disjoin
disjoining
disjoins
disjoint
disjointing
disjoints
disjunct
disjuncts
disk
disking
disks
disliking
dislimn
dislimning
dislimns
dislocating
dislocation
dislocations
dislodging
disloyal
disloyalty
dismal
dismally
dismals
dismantling
dismast
dismasting
dismasts
dismay
dismaying
dismays
dismiss
dismissal
dismissals
dismissing
dismount
dismounting
dismounts
disomic
disorganization
disorganizations
disorganizing
disown
disowning
disowns
disparaging
disparity
dispart
disparting
disparts
dispassion
dispassions
dispatch
dispatching
dispirit
dis

finfish
finfoot
finfoots
finial
finials
finical
finickin
finicky
finikin
finiking
fining
finings
finis
finish
finishing
fink
finking
finks
finky
finmark
finmarks
finnicky
finning
finnmark
finnmarks
finny
finochio
finochios
fins
fiord
fiords
fir
firing
firings
firkin
firkins
firm
firman
firmans
firming
firmly
firms
firn
firns
firry
firs
first
firstly
firsts
firth
firths
fisc
fiscal
fiscally
fiscals
fiscs
fish
fishboat
fishboats
fishbowl
fishbowls
fishgig
fishgigs
fishhook
fishhooks
fishily
fishing
fishings
fishpond
fishponds
fishtail
fishtailing
fishtails
fishway
fishways
fishy
fission
fissional
fissioning
fissions
fissuring
fist
fistful
fistfuls
fistic
fisticuffs
fisting
fists
fistula
fistular
fistulas
fisty
fit
fitch
fitchy
fitful
fitfully
fitly
fits
fitting
fittings
fix
fixatif
fixatifs
fixating
fixation
fixations
fixing
fixings
fixity
fixt
fiz
fizgig
fizgigs
fizz
fizzing
fizzling
fizzy
fjord
fjords
flab
flabbily
flabby
flabs
flaccid
flack
flacks
flacon
flacons
flag
flagging
flagging

gastral
gastric
gastrin
gastrins
gastronomic
gastronomical
gastronomy
gastrula
gastrulas
gasts
gasworks
gat
gating
gats
gaucho
gauchos
gaud
gaudily
gauds
gaudy
gauging
gault
gaults
gaum
gauming
gaums
gaun
gaunt
gauntly
gauntry
gaur
gaurs
gauss
gauzy
gavial
gavials
gavot
gavots
gavotting
gawk
gawkily
gawking
gawkish
gawks
gawky
gawsy
gay
gayal
gayals
gayly
gays
gaywing
gaywings
gazabo
gazabos
gazing
gazpacho
gazpachos
gharri
gharris
gharry
ghast
ghastful
ghastly
ghat
ghats
ghaut
ghauts
ghazi
ghazis
ghi
ghibli
ghiblis
ghis
ghost
ghosting
ghostly
ghosts
ghosty
ghoul
ghoulish
ghouls
ghyll
ghylls
giant
giantism
giantisms
giants
giaour
giaours
gib
gibbing
gibbon
gibbons
gibbous
gibing
gibingly
gibs
gid
giddap
giddily
giddy
giddying
gids
gift
gifting
gifts
gig
giga
gigabit
gigabits
gigantic
gigas
gigaton
gigatons
gigawatt
gigawatts
gigging
giggling
giggly
giglot
giglots
gigolo
gigolos
gigot
gigots
gigs
gild
gildhall
gildhalls
gilding
gildings
gilds
gill
gilling
gills
gilly
gillying
gilt
gilts

hobnob
hobnobbing
hobnobs
hobo
hoboing
hoboism
hoboisms
hobos
hobs
hock
hocking
hocks
hockshop
hockshops
hocus
hocusing
hocussing
hod
hodad
hodaddy
hodads
hoddin
hoddins
hods
hog
hogan
hogans
hogback
hogbacks
hogfish
hogg
hogging
hoggish
hoggs
hogmanay
hogmanays
hognut
hognuts
hogs
hogtying
hogwash
hoick
hoicking
hoicks
hoising
hoist
hoisting
hoists
hoking
hokku
hokum
hokums
hokypoky
holard
holards
hold
holdall
holdalls
holdback
holdbacks
holdfast
holdfasts
holding
holdings
holdout
holdouts
holds
holdup
holdups
holibut
holibuts
holiday
holidaying
holidays
holily
holing
holism
holisms
holist
holistic
holists
holk
holking
holks
holla
hollaing
holland
hollands
hollas
hollo
holloa
holloaing
holloas
holloing
holloo
hollooing
holloos
hollos
hollow
hollowing
hollowly
hollows
holly
hollyhock
hollyhocks
holm
holmic
holmium
holmiums
holms
holocaust
holocausts
hologram
holograms
hologyny
holozoic
holp
holt
holts
holy
holyday
holydays
homaging
homburg
homburgs
homicidal
homilist
homilists
homily
h

installation
installations
installing
installs
instals
instancing
instancy
instant
instantly
instants
instar
instarring
instars
instating
instigating
instigation
instigations
instigator
instigators
instil
instill
instilling
instills
instils
instinct
instincts
institution
institutional
institutionally
institutions
instruct
instructing
instruction
instructional
instructions
instructor
instructors
instructorship
instructorships
instructs
insubordination
insubordinations
insubstantial
insulant
insulants
insular
insularity
insulars
insulating
insulation
insulations
insulator
insulators
insulin
insulins
insult
insulting
insultingly
insults
insurant
insurants
insuring
insurmounably
inswathing
intact
intagli
intaglio
intaglios
intangibility
intangibly
intarsia
intarsias
inthral
inthrall
inthralling
inthralls
inthrals
inthroning
intima
intimacy
intimal
intimas
intimating
intimation
intimations
intimidating
intimidation
intimidations
intitling
intituling
into
intomb
intombing
intombs
intonating


limbi
limbic
limbing
limbo
limbos
limbs
limbus
limby
limina
liminal
liming
limit
limitary
limitation
limitations
limiting
limits
limn
limnic
limning
limns
limo
limos
limp
limpid
limpidly
limping
limpkin
limpkins
limply
limps
limpsy
limuli
limuloid
limuloids
limulus
limy
lin
linac
linacs
linalol
linalols
linalool
linalools
linchpin
linchpins
lindy
ling
linga
lingam
lingams
lingas
lingcod
lingcods
lingo
lings
lingua
lingual
linguals
linguini
linguinis
linguist
linguistic
linguistics
linguists
lingy
linin
lining
linings
linins
link
linkboy
linkboys
linking
linkman
links
linksman
linkup
linkups
linkwork
linkworks
linky
linn
linns
lino
linocut
linocuts
linos
lins
linsang
linsangs
linstock
linstocks
lint
lintol
lintols
lints
linty
linum
linums
liny
lion
lionfish
lionising
lionization
lionizations
lionizing
lions
lip
lipid
lipidic
lipids
lipin
lipins
lipoid
lipoidal
lipoids
lipoma
lipomas
lipomata
lipping
lippings
lippy
lips
lipstick
lipsticks
liquating
liquid
liquidating
liquidation
liquidat

miscuing
miscut
miscuts
miscutting
misdating
misdid
misdo
misdoing
misdoings
misdoubt
misdoubting
misdoubts
misdraw
misdrawing
misdrawn
misdraws
misdriving
misfaith
misfaiths
misfiling
misfiring
misfit
misfits
misfitting
misform
misforming
misforms
misframing
misgauging
misgiving
misgivings
misgraft
misgrafting
misgrafts
misgrow
misgrowing
misgrown
misgrows
misguiding
mishap
mishaps
mishit
mishits
mishitting
mishmash
mishmosh
misinform
misinformation
misinformations
misinforms
misjoin
misjoining
misjoins
misjudging
miskal
miskals
misknow
misknowing
misknown
misknows
mislabor
mislaboring
mislabors
mislaid
mislain
mislay
mislaying
mislays
mislight
mislighting
mislights
misliking
mislit
misliving
mislodging
mislying
mismanaging
mismark
mismarking
mismarks
mismatch
mismatching
mismating
mismoving
misnaming
miso
misogamy
misogynist
misogynists
misogyny
misology
misos
mispaging
mispaint
mispainting
mispaints
misparsing
mispart
misparting
misparts
mispatch
mispatching
misplacing
misplant
misp

nit
nitid
niton
nitons
nitpick
nitpicking
nitpicks
nitrating
nitrator
nitrators
nitric
nitrid
nitrids
nitrify
nitrifying
nitril
nitrils
nitro
nitrolic
nitros
nitroso
nitrosyl
nitrosyls
nitrous
nits
nitty
nitwit
nitwits
nival
nix
nixing
nixy
nizam
nizams
no
nob
nobbily
nobbling
nobby
nobility
nobly
nobody
nobs
nock
nocking
nocks
noctuid
noctuids
noctuoid
nocturn
nocturnal
nocturns
nocuous
nod
nodal
nodality
nodally
nodding
noddling
noddy
nodi
nodical
nodosity
nodous
nods
nodular
nodulous
nodus
nog
nogg
noggin
nogging
noggings
noggins
noggs
nogs
noh
nohow
noil
noils
noily
noir
noisily
noising
noisy
nolo
nolos
nom
noma
nomad
nomadic
nomadism
nomadisms
nomads
nomarch
nomarchs
nomarchy
nomas
nombril
nombrils
nomina
nominal
nominally
nominals
nominating
nomination
nominations
nomism
nomisms
nomistic
nomogram
nomograms
nomoi
nomology
nomos
noms
nona
nonacid
nonacids
nonadult
nonadults
nonagon
nonagons
nonalcoholic
nonas
nonautomatic
nonbank
nonbasic
nonbook
nonbooks
noncash
nonchalant
nonchal

outpaint
outpainting
outpaints
outpass
outpassing
outpity
outpitying
outplan
outplanning
outplans
outplay
outplaying
outplays
outplod
outplodding
outplods
outpoint
outpointing
outpoints
outpoll
outpolling
outpolls
outport
outports
outpost
outposts
outpour
outpouring
outpours
outpray
outpraying
outprays
outpricing
outpull
outpulling
outpulls
outpush
outpushing
output
outputs
outputting
outquoting
outracing
outraging
outraising
outran
outrang
outranging
outrank
outranking
outranks
outraving
outriding
outright
outring
outringing
outrings
outrival
outrivaling
outrivalling
outrivals
outroar
outroaring
outroars
outrock
outrocking
outrocks
outroll
outrolling
outrolls
outroot
outrooting
outroots
outrun
outrung
outrunning
outruns
outrush
outs
outsail
outsailing
outsails
outsang
outsat
outsavor
outsavoring
outsavors
outsaw
outscold
outscolding
outscolds
outscoring
outscorn
outscorning
outscorns
outshaming
outshining
outshoot
outshooting
outshoots
outshot
outshout
outshouting
outshouts
outsight
o

plop
plopping
plops
plosion
plosions
plot
plots
plotting
plotty
plough
ploughing
ploughs
plow
plowback
plowbacks
plowboy
plowboys
plowing
plowland
plowlands
plowman
plows
ploy
ploying
ploys
pluck
pluckily
plucking
plucks
plucky
plug
plugging
plugs
plugugly
plum
plumb
plumbago
plumbagos
plumbic
plumbing
plumbings
plumbism
plumbisms
plumbous
plumbs
plumbum
plumbums
pluming
plummy
plump
plumping
plumpish
plumply
plumps
plums
plumular
plumy
plunging
plunk
plunking
plunks
plural
pluralism
plurality
pluralization
pluralizations
pluralizing
plurally
plurals
plus
plush
plushily
plushly
plushy
plutocracy
plutocrat
plutocratic
plutocrats
pluton
plutonic
plutonium
plutoniums
plutons
pluvial
pluvials
pluvious
ply
plying
plyingly
plywood
plywoods
poach
poaching
poachy
pochard
pochards
pock
pockily
pocking
pockmark
pockmarking
pockmarks
pocks
pocky
poco
pocosin
pocosins
pod
podagra
podagral
podagras
podagric
podding
podgily
podgy
podia
podiatrist
podiatry
poditic
podium
podiums
pods
podsol
podsolic


raffling
raffs
raft
rafting
rafts
raftsman
rag
raga
ragamuffin
ragamuffins
ragas
ragbag
ragbags
ragging
raggy
ragi
raging
ragingly
ragis
raglan
raglans
ragman
ragout
ragouting
ragouts
rags
ragtag
ragtags
ragwort
ragworts
rah
raia
raias
raid
raiding
raids
rail
railbird
railbirds
railing
railings
railroad
railroading
railroadings
railroads
rails
railway
railways
rain
rainband
rainbands
rainbird
rainbirds
rainbow
rainbows
raincoat
raincoats
raindrop
raindrops
rainfall
rainfalls
rainily
raining
rainmaking
rainmakings
rainout
rainouts
rains
rainstorm
rainstorms
rainwash
rainy
raisin
raising
raisings
raisins
raisiny
raj
raja
rajah
rajahs
rajas
raki
raking
rakis
rakish
rakishly
rally
rallying
rallyings
rallyist
rallyists
ram
rambling
rambunctious
rambutan
rambutans
rami
ramification
ramifications
ramiform
ramify
ramifying
ramming
rammish
rammy
ramosity
ramous
ramp
rampaging
rampancy
rampant
rampantly
rampart
ramparting
ramparts
ramping
rampion
rampions
ramps
ramrod
ramrods
rams
ramshorn
ramsh

sarodists
sarods
sarong
sarongs
sarsaparilla
sarsaparillas
sarsar
sarsars
sartor
sartorii
sartors
sash
sashay
sashaying
sashays
sashimi
sashimis
sashing
sasin
sasins
sass
sassaby
sassafras
sassily
sassing
sasswood
sasswoods
sassy
sastruga
sastrugi
sat
satang
satangs
satanic
satanism
satanisms
satanist
satanists
satara
sataras
sati
satiably
satiating
satin
sating
satinpod
satinpods
satins
satiny
satiric
satirical
satirically
satirising
satirist
satirists
satirizing
satis
satisfaction
satisfactions
satisfactorily
satisfactory
satisfy
satisfying
satisfyingly
satori
satoris
satrap
satraps
satrapy
saturant
saturants
saturating
saturation
saturations
satyr
satyric
satyrid
satyrids
satyrs
sau
sauch
sauchs
saucily
saucing
saucy
saugh
saughs
saughy
saul
sauls
sault
saults
sauna
saunas
saurian
saurians
sauropod
sauropods
saury
sautoir
sautoirs
savaging
savagism
savagisms
savanna
savannah
savannahs
savannas
savant
savants
savin
saving
savingly
savings
savins
savior
saviors
saviour
saviours
savor


solarism
solarisms
solarium
solariums
solarizing
solatia
solating
solation
solations
solatium
sold
soldan
soldans
soldi
soldo
soli
solicit
solicitation
soliciting
solicitor
solicitors
solicitous
solicits
solid
solidago
solidagos
solidarity
solidary
solidi
solidification
solidifications
solidify
solidifying
solidity
solidly
solids
solidus
soliloquizing
soliloquy
soliloquys
soling
solion
solions
soliquid
soliquids
solitary
solo
soloing
soloist
soloists
solon
solons
solos
sols
solubility
solubly
solum
solums
solus
solution
solutions
solvating
solving
soma
somas
somata
somatic
sombrous
somital
somitic
somnambulism
somnambulist
somnambulists
son
sonant
sonantal
sonantic
sonants
sonar
sonarman
sonars
sonata
sonatas
sonatina
sonatinas
song
songbird
songbirds
songbook
songbooks
songful
songs
sonic
sonicating
sonics
sonly
sonny
sonorant
sonorants
sonority
sonorous
sonovox
sons
sonship
sonships
sonsy
soochong
soochongs
soon
soot
sooth
soothing
soothly
sooths
soothsaid
soothsay
soothsaying
sooths

swizzling
swob
swobbing
swobs
swoon
swooning
swoons
swoop
swooping
swoops
swoosh
swooshing
swop
swopping
swops
sword
swordfish
swordman
swords
sworn
swot
swots
swotting
swoun
swound
swounding
swounds
swouning
swouns
swum
swung
sybo
syconia
syconium
sycophant
sycophantic
sycophants
sycosis
syllabi
syllabic
syllabics
syllabling
syllabub
syllabubs
syllabus
sylph
sylphic
sylphid
sylphids
sylphish
sylphs
sylphy
sylva
sylvan
sylvans
sylvas
sylvatic
sylvin
sylvins
symbion
symbions
symbiont
symbionts
symbiot
symbiots
symbol
symbolic
symbolical
symbolically
symboling
symbolism
symbolisms
symbolization
symbolizations
symbolizing
symbolling
symbols
sympathizing
sympathy
sympatry
symphonic
symphony
sympodia
symposia
symposium
symptom
symptomatically
symptomatology
symptoms
syn
synagog
synagogs
synapsing
synapsis
synaptic
sync
syncarp
syncarps
syncarpy
synch
synching
synchro
synchronization
synchronizations
synchronizing
synchros
synchs
syncing
syncom
syncoms
syncopal
syncopating
syncopation
syncop

tsarist
tsarists
tsaritza
tsaritzas
tsars
tsk
tsking
tsks
tsktsk
tsktsking
tsktsks
tsuba
tsunami
tsunamic
tsunamis
tsuris
tuatara
tuataras
tub
tuba
tubal
tubas
tubbing
tubby
tubful
tubfuls
tubiform
tubing
tubings
tubs
tubular
tubulating
tubulous
tuchun
tuchuns
tuck
tucking
tucks
tufa
tufas
tuff
tuffs
tuft
tuftily
tufting
tufts
tufty
tug
tugboat
tugboats
tugging
tugrik
tugriks
tugs
tui
tuis
tuition
tuitions
tuladi
tuladis
tulip
tulips
tumbling
tumblings
tumbril
tumbrils
tumid
tumidily
tumidity
tummy
tumor
tumoral
tumorous
tumors
tumour
tumours
tump
tumps
tumular
tumuli
tumulous
tumult
tumults
tumultuous
tumulus
tun
tuna
tunably
tunas
tundish
tundra
tundras
tung
tungs
tungstic
tunic
tunica
tunics
tuning
tunning
tunny
tuns
tup
tupik
tupiks
tupping
tups
turaco
turacos
turacou
turacous
turban
turbans
turbary
turbid
turbidity
turbidly
turbinal
turbinals
turbit
turbith
turbiths
turbits
turbo
turbocar
turbocars
turbofan
turbofans
turboprop
turboprops
turbos
turbot
turbots
turd
turds
turf
turfi

waff
waffing
waffling
waffs
waft
wafting
wafts
wag
wagging
waggish
waggling
waggly
waggon
waggoning
waggons
waging
wagon
wagoning
wagons
wags
wagtail
wagtails
wahconda
wahcondas
wahoo
wahoos
waif
waifing
waifs
wail
wailful
wailing
wails
wain
wains
wainscot
wainscoting
wainscots
wainscotting
wair
wairing
wairs
waist
waisting
waistings
waists
wait
waiting
waitings
waits
waiving
wakanda
wakandas
wakiki
wakikis
waking
waling
walk
walkaway
walkaways
walking
walkings
walkout
walkouts
walks
walkup
walkups
walkway
walkways
wall
walla
wallaby
wallah
wallahs
wallaroo
wallaroos
wallas
walling
wallop
walloping
wallops
wallow
wallowing
wallows
walls
wally
walnut
walnuts
walrus
waltz
waltzing
waly
wambling
wambly
wammus
wampish
wampishing
wampum
wampums
wampus
wamus
wan
wand
wands
wangan
wangans
wangling
wangun
wanguns
wanigan
wanigans
waning
wanion
wanions
wanly
wannigan
wannigans
wanning
wans
want
wanting
wanton
wantoning
wantonly
wantons
wants
wany
wap
wapiti
wapitis
wapping
waps
war
warbling
war

37641

In [11]:
# Tallying number of words in words.txt

fin = open('words.txt')
tally = 0
for line in fin:
    tally += 1

# Calculating and printing the percentage of words without 'e'
    
print("{0:.6}% of the words in 'words.txt' do not have the letter 'e'.".format((no_e_tally/tally) * 100))

33.0738% of the words in 'words.txt' do not have the letter 'e'.


#### Exercise 3  
*Write a function named `avoids` that takes a word and a string of forbidden letters, and that returns `True` if the word doesn’t use any of the forbidden letters.*

*Write a program that prompts the user to enter a string of forbidden letters and then prints the number of words that don’t contain any of them. Can you find a combination of 5 forbidden letters that excludes the smallest number of words?*


In [12]:
def avoids(forbidden, word):
    """
    Returns False if word contains one of 
    the characters in the forbidden string.
    """
    for character in forbidden:
        if character in word:
            return False
    return True

In [13]:
forbidden = "abcde"
word = "grape"
avoids(forbidden, word)

False

In [14]:
forbidden = "abcde"
word = "stuff"
avoids(forbidden, word)

True

*__There are $26 \choose 5$ - or 65,780 - different ways of choosing five letters from the alphabet. The most efficient way of selecting the five letters that exclude the fewest words might be to look at a list of letter frequency (e.g., [this one from Wikipedia](https://en.wikipedia.org/wiki/Letter_frequency "letter frequency")), and choose the five least common letters:__*

In [15]:
# Tallying and printing words in words.txt without 'zqxjk'

fin = open('words.txt')
forbidden = "zqxjk"
avoids_tally = tally
for line in fin:
    if avoids(forbidden, line.strip()):
        avoids_tally -= 1
    
avoids_tally

17945

*__While this may seem a logical way to approach the problem, it unfortunately does not generate the correct answer.  For the sake of completeness, I created an algorithm that generated all 65,780 possible five-letter combinations, and tested all of them with the function `avoids`.  The algorithm that generated the combinations is fairly verbose, but it uses only functions that have been introduced in the book so far:__*

In [16]:
# create a string with all of the letters of the alphabet

alphabet = ""
for i in range(ord('a'), ord('z') + 1):
    alphabet += chr(i)
    
# Use that string to generate all the possible combinations

n = 0
p1 = n
while p1 < len(alphabet):
    p2 = p1 + 1
    while p2 < len(alphabet):
        p3 = p2 + 1
        while p3 < len(alphabet):
            p4 = p3 + 1
            while p4 < len(alphabet):
                p5 = p4 + 1
                while p5 < len(alphabet):
                    letter_seq = (alphabet[p1] + alphabet[p2] + alphabet[p3] +
                                 alphabet[p4] + alphabet[p5])
                    print(letter_seq)
                    p5 += 1
                p4 += 1
            p3 += 1
        p2 += 1
    p1 += 1
    


abcde
abcdf
abcdg
abcdh
abcdi
abcdj
abcdk
abcdl
abcdm
abcdn
abcdo
abcdp
abcdq
abcdr
abcds
abcdt
abcdu
abcdv
abcdw
abcdx
abcdy
abcdz
abcef
abceg
abceh
abcei
abcej
abcek
abcel
abcem
abcen
abceo
abcep
abceq
abcer
abces
abcet
abceu
abcev
abcew
abcex
abcey
abcez
abcfg
abcfh
abcfi
abcfj
abcfk
abcfl
abcfm
abcfn
abcfo
abcfp
abcfq
abcfr
abcfs
abcft
abcfu
abcfv
abcfw
abcfx
abcfy
abcfz
abcgh
abcgi
abcgj
abcgk
abcgl
abcgm
abcgn
abcgo
abcgp
abcgq
abcgr
abcgs
abcgt
abcgu
abcgv
abcgw
abcgx
abcgy
abcgz
abchi
abchj
abchk
abchl
abchm
abchn
abcho
abchp
abchq
abchr
abchs
abcht
abchu
abchv
abchw
abchx
abchy
abchz
abcij
abcik
abcil
abcim
abcin
abcio
abcip
abciq
abcir
abcis
abcit
abciu
abciv
abciw
abcix
abciy
abciz
abcjk
abcjl
abcjm
abcjn
abcjo
abcjp
abcjq
abcjr
abcjs
abcjt
abcju
abcjv
abcjw
abcjx
abcjy
abcjz
abckl
abckm
abckn
abcko
abckp
abckq
abckr
abcks
abckt
abcku
abckv
abckw
abckx
abcky
abckz
abclm
abcln
abclo
abclp
abclq
abclr
abcls
abclt
abclu
abclv
abclw
abclx
abcly
abclz
abcmn
abcmo
abcmp
abcmq
abcm

ablqx
ablqy
ablqz
ablrs
ablrt
ablru
ablrv
ablrw
ablrx
ablry
ablrz
ablst
ablsu
ablsv
ablsw
ablsx
ablsy
ablsz
abltu
abltv
abltw
abltx
ablty
abltz
abluv
abluw
ablux
abluy
abluz
ablvw
ablvx
ablvy
ablvz
ablwx
ablwy
ablwz
ablxy
ablxz
ablyz
abmno
abmnp
abmnq
abmnr
abmns
abmnt
abmnu
abmnv
abmnw
abmnx
abmny
abmnz
abmop
abmoq
abmor
abmos
abmot
abmou
abmov
abmow
abmox
abmoy
abmoz
abmpq
abmpr
abmps
abmpt
abmpu
abmpv
abmpw
abmpx
abmpy
abmpz
abmqr
abmqs
abmqt
abmqu
abmqv
abmqw
abmqx
abmqy
abmqz
abmrs
abmrt
abmru
abmrv
abmrw
abmrx
abmry
abmrz
abmst
abmsu
abmsv
abmsw
abmsx
abmsy
abmsz
abmtu
abmtv
abmtw
abmtx
abmty
abmtz
abmuv
abmuw
abmux
abmuy
abmuz
abmvw
abmvx
abmvy
abmvz
abmwx
abmwy
abmwz
abmxy
abmxz
abmyz
abnop
abnoq
abnor
abnos
abnot
abnou
abnov
abnow
abnox
abnoy
abnoz
abnpq
abnpr
abnps
abnpt
abnpu
abnpv
abnpw
abnpx
abnpy
abnpz
abnqr
abnqs
abnqt
abnqu
abnqv
abnqw
abnqx
abnqy
abnqz
abnrs
abnrt
abnru
abnrv
abnrw
abnrx
abnry
abnrz
abnst
abnsu
abnsv
abnsw
abnsx
abnsy
abnsz
abntu
abntv
abntw
abntx
abnt

acjkr
acjks
acjkt
acjku
acjkv
acjkw
acjkx
acjky
acjkz
acjlm
acjln
acjlo
acjlp
acjlq
acjlr
acjls
acjlt
acjlu
acjlv
acjlw
acjlx
acjly
acjlz
acjmn
acjmo
acjmp
acjmq
acjmr
acjms
acjmt
acjmu
acjmv
acjmw
acjmx
acjmy
acjmz
acjno
acjnp
acjnq
acjnr
acjns
acjnt
acjnu
acjnv
acjnw
acjnx
acjny
acjnz
acjop
acjoq
acjor
acjos
acjot
acjou
acjov
acjow
acjox
acjoy
acjoz
acjpq
acjpr
acjps
acjpt
acjpu
acjpv
acjpw
acjpx
acjpy
acjpz
acjqr
acjqs
acjqt
acjqu
acjqv
acjqw
acjqx
acjqy
acjqz
acjrs
acjrt
acjru
acjrv
acjrw
acjrx
acjry
acjrz
acjst
acjsu
acjsv
acjsw
acjsx
acjsy
acjsz
acjtu
acjtv
acjtw
acjtx
acjty
acjtz
acjuv
acjuw
acjux
acjuy
acjuz
acjvw
acjvx
acjvy
acjvz
acjwx
acjwy
acjwz
acjxy
acjxz
acjyz
acklm
ackln
acklo
acklp
acklq
acklr
ackls
acklt
acklu
acklv
acklw
acklx
ackly
acklz
ackmn
ackmo
ackmp
ackmq
ackmr
ackms
ackmt
ackmu
ackmv
ackmw
ackmx
ackmy
ackmz
ackno
acknp
acknq
acknr
ackns
acknt
acknu
acknv
acknw
acknx
ackny
acknz
ackop
ackoq
ackor
ackos
ackot
ackou
ackov
ackow
ackox
ackoy
ackoz
ackpq
ackpr
ackp

adopv
adopw
adopx
adopy
adopz
adoqr
adoqs
adoqt
adoqu
adoqv
adoqw
adoqx
adoqy
adoqz
adors
adort
adoru
adorv
adorw
adorx
adory
adorz
adost
adosu
adosv
adosw
adosx
adosy
adosz
adotu
adotv
adotw
adotx
adoty
adotz
adouv
adouw
adoux
adouy
adouz
adovw
adovx
adovy
adovz
adowx
adowy
adowz
adoxy
adoxz
adoyz
adpqr
adpqs
adpqt
adpqu
adpqv
adpqw
adpqx
adpqy
adpqz
adprs
adprt
adpru
adprv
adprw
adprx
adpry
adprz
adpst
adpsu
adpsv
adpsw
adpsx
adpsy
adpsz
adptu
adptv
adptw
adptx
adpty
adptz
adpuv
adpuw
adpux
adpuy
adpuz
adpvw
adpvx
adpvy
adpvz
adpwx
adpwy
adpwz
adpxy
adpxz
adpyz
adqrs
adqrt
adqru
adqrv
adqrw
adqrx
adqry
adqrz
adqst
adqsu
adqsv
adqsw
adqsx
adqsy
adqsz
adqtu
adqtv
adqtw
adqtx
adqty
adqtz
adquv
adquw
adqux
adquy
adquz
adqvw
adqvx
adqvy
adqvz
adqwx
adqwy
adqwz
adqxy
adqxz
adqyz
adrst
adrsu
adrsv
adrsw
adrsx
adrsy
adrsz
adrtu
adrtv
adrtw
adrtx
adrty
adrtz
adruv
adruw
adrux
adruy
adruz
adrvw
adrvx
adrvy
adrvz
adrwx
adrwy
adrwz
adrxy
adrxz
adryz
adstu
adstv
adstw
adstx
adsty
adstz
adsuv
adsu

afiwy
afiwz
afixy
afixz
afiyz
afjkl
afjkm
afjkn
afjko
afjkp
afjkq
afjkr
afjks
afjkt
afjku
afjkv
afjkw
afjkx
afjky
afjkz
afjlm
afjln
afjlo
afjlp
afjlq
afjlr
afjls
afjlt
afjlu
afjlv
afjlw
afjlx
afjly
afjlz
afjmn
afjmo
afjmp
afjmq
afjmr
afjms
afjmt
afjmu
afjmv
afjmw
afjmx
afjmy
afjmz
afjno
afjnp
afjnq
afjnr
afjns
afjnt
afjnu
afjnv
afjnw
afjnx
afjny
afjnz
afjop
afjoq
afjor
afjos
afjot
afjou
afjov
afjow
afjox
afjoy
afjoz
afjpq
afjpr
afjps
afjpt
afjpu
afjpv
afjpw
afjpx
afjpy
afjpz
afjqr
afjqs
afjqt
afjqu
afjqv
afjqw
afjqx
afjqy
afjqz
afjrs
afjrt
afjru
afjrv
afjrw
afjrx
afjry
afjrz
afjst
afjsu
afjsv
afjsw
afjsx
afjsy
afjsz
afjtu
afjtv
afjtw
afjtx
afjty
afjtz
afjuv
afjuw
afjux
afjuy
afjuz
afjvw
afjvx
afjvy
afjvz
afjwx
afjwy
afjwz
afjxy
afjxz
afjyz
afklm
afkln
afklo
afklp
afklq
afklr
afkls
afklt
afklu
afklv
afklw
afklx
afkly
afklz
afkmn
afkmo
afkmp
afkmq
afkmr
afkms
afkmt
afkmu
afkmv
afkmw
afkmx
afkmy
afkmz
afkno
afknp
afknq
afknr
afkns
afknt
afknu
afknv
afknw
afknx
afkny
afknz
afkop
afkoq
afko

ahktz
ahkuv
ahkuw
ahkux
ahkuy
ahkuz
ahkvw
ahkvx
ahkvy
ahkvz
ahkwx
ahkwy
ahkwz
ahkxy
ahkxz
ahkyz
ahlmn
ahlmo
ahlmp
ahlmq
ahlmr
ahlms
ahlmt
ahlmu
ahlmv
ahlmw
ahlmx
ahlmy
ahlmz
ahlno
ahlnp
ahlnq
ahlnr
ahlns
ahlnt
ahlnu
ahlnv
ahlnw
ahlnx
ahlny
ahlnz
ahlop
ahloq
ahlor
ahlos
ahlot
ahlou
ahlov
ahlow
ahlox
ahloy
ahloz
ahlpq
ahlpr
ahlps
ahlpt
ahlpu
ahlpv
ahlpw
ahlpx
ahlpy
ahlpz
ahlqr
ahlqs
ahlqt
ahlqu
ahlqv
ahlqw
ahlqx
ahlqy
ahlqz
ahlrs
ahlrt
ahlru
ahlrv
ahlrw
ahlrx
ahlry
ahlrz
ahlst
ahlsu
ahlsv
ahlsw
ahlsx
ahlsy
ahlsz
ahltu
ahltv
ahltw
ahltx
ahlty
ahltz
ahluv
ahluw
ahlux
ahluy
ahluz
ahlvw
ahlvx
ahlvy
ahlvz
ahlwx
ahlwy
ahlwz
ahlxy
ahlxz
ahlyz
ahmno
ahmnp
ahmnq
ahmnr
ahmns
ahmnt
ahmnu
ahmnv
ahmnw
ahmnx
ahmny
ahmnz
ahmop
ahmoq
ahmor
ahmos
ahmot
ahmou
ahmov
ahmow
ahmox
ahmoy
ahmoz
ahmpq
ahmpr
ahmps
ahmpt
ahmpu
ahmpv
ahmpw
ahmpx
ahmpy
ahmpz
ahmqr
ahmqs
ahmqt
ahmqu
ahmqv
ahmqw
ahmqx
ahmqy
ahmqz
ahmrs
ahmrt
ahmru
ahmrv
ahmrw
ahmrx
ahmry
ahmrz
ahmst
ahmsu
ahmsv
ahmsw
ahmsx
ahmsy
ahmsz
ahmtu
ahmtv
ahmt

akoyz
akpqr
akpqs
akpqt
akpqu
akpqv
akpqw
akpqx
akpqy
akpqz
akprs
akprt
akpru
akprv
akprw
akprx
akpry
akprz
akpst
akpsu
akpsv
akpsw
akpsx
akpsy
akpsz
akptu
akptv
akptw
akptx
akpty
akptz
akpuv
akpuw
akpux
akpuy
akpuz
akpvw
akpvx
akpvy
akpvz
akpwx
akpwy
akpwz
akpxy
akpxz
akpyz
akqrs
akqrt
akqru
akqrv
akqrw
akqrx
akqry
akqrz
akqst
akqsu
akqsv
akqsw
akqsx
akqsy
akqsz
akqtu
akqtv
akqtw
akqtx
akqty
akqtz
akquv
akquw
akqux
akquy
akquz
akqvw
akqvx
akqvy
akqvz
akqwx
akqwy
akqwz
akqxy
akqxz
akqyz
akrst
akrsu
akrsv
akrsw
akrsx
akrsy
akrsz
akrtu
akrtv
akrtw
akrtx
akrty
akrtz
akruv
akruw
akrux
akruy
akruz
akrvw
akrvx
akrvy
akrvz
akrwx
akrwy
akrwz
akrxy
akrxz
akryz
akstu
akstv
akstw
akstx
aksty
akstz
aksuv
aksuw
aksux
aksuy
aksuz
aksvw
aksvx
aksvy
aksvz
akswx
akswy
akswz
aksxy
aksxz
aksyz
aktuv
aktuw
aktux
aktuy
aktuz
aktvw
aktvx
aktvy
aktvz
aktwx
aktwy
aktwz
aktxy
aktxz
aktyz
akuvw
akuvx
akuvy
akuvz
akuwx
akuwy
akuwz
akuxy
akuxz
akuyz
akvwx
akvwy
akvwz
akvxy
akvxz
akvyz
akwxy
akwxz
akwyz
akxyz
almn

bcfhq
bcfhr
bcfhs
bcfht
bcfhu
bcfhv
bcfhw
bcfhx
bcfhy
bcfhz
bcfij
bcfik
bcfil
bcfim
bcfin
bcfio
bcfip
bcfiq
bcfir
bcfis
bcfit
bcfiu
bcfiv
bcfiw
bcfix
bcfiy
bcfiz
bcfjk
bcfjl
bcfjm
bcfjn
bcfjo
bcfjp
bcfjq
bcfjr
bcfjs
bcfjt
bcfju
bcfjv
bcfjw
bcfjx
bcfjy
bcfjz
bcfkl
bcfkm
bcfkn
bcfko
bcfkp
bcfkq
bcfkr
bcfks
bcfkt
bcfku
bcfkv
bcfkw
bcfkx
bcfky
bcfkz
bcflm
bcfln
bcflo
bcflp
bcflq
bcflr
bcfls
bcflt
bcflu
bcflv
bcflw
bcflx
bcfly
bcflz
bcfmn
bcfmo
bcfmp
bcfmq
bcfmr
bcfms
bcfmt
bcfmu
bcfmv
bcfmw
bcfmx
bcfmy
bcfmz
bcfno
bcfnp
bcfnq
bcfnr
bcfns
bcfnt
bcfnu
bcfnv
bcfnw
bcfnx
bcfny
bcfnz
bcfop
bcfoq
bcfor
bcfos
bcfot
bcfou
bcfov
bcfow
bcfox
bcfoy
bcfoz
bcfpq
bcfpr
bcfps
bcfpt
bcfpu
bcfpv
bcfpw
bcfpx
bcfpy
bcfpz
bcfqr
bcfqs
bcfqt
bcfqu
bcfqv
bcfqw
bcfqx
bcfqy
bcfqz
bcfrs
bcfrt
bcfru
bcfrv
bcfrw
bcfrx
bcfry
bcfrz
bcfst
bcfsu
bcfsv
bcfsw
bcfsx
bcfsy
bcfsz
bcftu
bcftv
bcftw
bcftx
bcfty
bcftz
bcfuv
bcfuw
bcfux
bcfuy
bcfuz
bcfvw
bcfvx
bcfvy
bcfvz
bcfwx
bcfwy
bcfwz
bcfxy
bcfxz
bcfyz
bcghi
bcghj
bcghk
bcgh

bdhsu
bdhsv
bdhsw
bdhsx
bdhsy
bdhsz
bdhtu
bdhtv
bdhtw
bdhtx
bdhty
bdhtz
bdhuv
bdhuw
bdhux
bdhuy
bdhuz
bdhvw
bdhvx
bdhvy
bdhvz
bdhwx
bdhwy
bdhwz
bdhxy
bdhxz
bdhyz
bdijk
bdijl
bdijm
bdijn
bdijo
bdijp
bdijq
bdijr
bdijs
bdijt
bdiju
bdijv
bdijw
bdijx
bdijy
bdijz
bdikl
bdikm
bdikn
bdiko
bdikp
bdikq
bdikr
bdiks
bdikt
bdiku
bdikv
bdikw
bdikx
bdiky
bdikz
bdilm
bdiln
bdilo
bdilp
bdilq
bdilr
bdils
bdilt
bdilu
bdilv
bdilw
bdilx
bdily
bdilz
bdimn
bdimo
bdimp
bdimq
bdimr
bdims
bdimt
bdimu
bdimv
bdimw
bdimx
bdimy
bdimz
bdino
bdinp
bdinq
bdinr
bdins
bdint
bdinu
bdinv
bdinw
bdinx
bdiny
bdinz
bdiop
bdioq
bdior
bdios
bdiot
bdiou
bdiov
bdiow
bdiox
bdioy
bdioz
bdipq
bdipr
bdips
bdipt
bdipu
bdipv
bdipw
bdipx
bdipy
bdipz
bdiqr
bdiqs
bdiqt
bdiqu
bdiqv
bdiqw
bdiqx
bdiqy
bdiqz
bdirs
bdirt
bdiru
bdirv
bdirw
bdirx
bdiry
bdirz
bdist
bdisu
bdisv
bdisw
bdisx
bdisy
bdisz
bditu
bditv
bditw
bditx
bdity
bditz
bdiuv
bdiuw
bdiux
bdiuy
bdiuz
bdivw
bdivx
bdivy
bdivz
bdiwx
bdiwy
bdiwz
bdixy
bdixz
bdiyz
bdjkl
bdjkm
bdjkn
bdjk

beovx
beovy
beovz
beowx
beowy
beowz
beoxy
beoxz
beoyz
bepqr
bepqs
bepqt
bepqu
bepqv
bepqw
bepqx
bepqy
bepqz
beprs
beprt
bepru
beprv
beprw
beprx
bepry
beprz
bepst
bepsu
bepsv
bepsw
bepsx
bepsy
bepsz
beptu
beptv
beptw
beptx
bepty
beptz
bepuv
bepuw
bepux
bepuy
bepuz
bepvw
bepvx
bepvy
bepvz
bepwx
bepwy
bepwz
bepxy
bepxz
bepyz
beqrs
beqrt
beqru
beqrv
beqrw
beqrx
beqry
beqrz
beqst
beqsu
beqsv
beqsw
beqsx
beqsy
beqsz
beqtu
beqtv
beqtw
beqtx
beqty
beqtz
bequv
bequw
bequx
bequy
bequz
beqvw
beqvx
beqvy
beqvz
beqwx
beqwy
beqwz
beqxy
beqxz
beqyz
berst
bersu
bersv
bersw
bersx
bersy
bersz
bertu
bertv
bertw
bertx
berty
bertz
beruv
beruw
berux
beruy
beruz
bervw
bervx
bervy
bervz
berwx
berwy
berwz
berxy
berxz
beryz
bestu
bestv
bestw
bestx
besty
bestz
besuv
besuw
besux
besuy
besuz
besvw
besvx
besvy
besvz
beswx
beswy
beswz
besxy
besxz
besyz
betuv
betuw
betux
betuy
betuz
betvw
betvx
betvy
betvz
betwx
betwy
betwz
betxy
betxz
betyz
beuvw
beuvx
beuvy
beuvz
beuwx
beuwy
beuwz
beuxy
beuxz
beuyz
bevwx
bevwy
bevw

bgkmq
bgkmr
bgkms
bgkmt
bgkmu
bgkmv
bgkmw
bgkmx
bgkmy
bgkmz
bgkno
bgknp
bgknq
bgknr
bgkns
bgknt
bgknu
bgknv
bgknw
bgknx
bgkny
bgknz
bgkop
bgkoq
bgkor
bgkos
bgkot
bgkou
bgkov
bgkow
bgkox
bgkoy
bgkoz
bgkpq
bgkpr
bgkps
bgkpt
bgkpu
bgkpv
bgkpw
bgkpx
bgkpy
bgkpz
bgkqr
bgkqs
bgkqt
bgkqu
bgkqv
bgkqw
bgkqx
bgkqy
bgkqz
bgkrs
bgkrt
bgkru
bgkrv
bgkrw
bgkrx
bgkry
bgkrz
bgkst
bgksu
bgksv
bgksw
bgksx
bgksy
bgksz
bgktu
bgktv
bgktw
bgktx
bgkty
bgktz
bgkuv
bgkuw
bgkux
bgkuy
bgkuz
bgkvw
bgkvx
bgkvy
bgkvz
bgkwx
bgkwy
bgkwz
bgkxy
bgkxz
bgkyz
bglmn
bglmo
bglmp
bglmq
bglmr
bglms
bglmt
bglmu
bglmv
bglmw
bglmx
bglmy
bglmz
bglno
bglnp
bglnq
bglnr
bglns
bglnt
bglnu
bglnv
bglnw
bglnx
bglny
bglnz
bglop
bgloq
bglor
bglos
bglot
bglou
bglov
bglow
bglox
bgloy
bgloz
bglpq
bglpr
bglps
bglpt
bglpu
bglpv
bglpw
bglpx
bglpy
bglpz
bglqr
bglqs
bglqt
bglqu
bglqv
bglqw
bglqx
bglqy
bglqz
bglrs
bglrt
bglru
bglrv
bglrw
bglrx
bglry
bglrz
bglst
bglsu
bglsv
bglsw
bglsx
bglsy
bglsz
bgltu
bgltv
bgltw
bgltx
bglty
bgltz
bgluv
bgluw
bglu

bikvy
bikvz
bikwx
bikwy
bikwz
bikxy
bikxz
bikyz
bilmn
bilmo
bilmp
bilmq
bilmr
bilms
bilmt
bilmu
bilmv
bilmw
bilmx
bilmy
bilmz
bilno
bilnp
bilnq
bilnr
bilns
bilnt
bilnu
bilnv
bilnw
bilnx
bilny
bilnz
bilop
biloq
bilor
bilos
bilot
bilou
bilov
bilow
bilox
biloy
biloz
bilpq
bilpr
bilps
bilpt
bilpu
bilpv
bilpw
bilpx
bilpy
bilpz
bilqr
bilqs
bilqt
bilqu
bilqv
bilqw
bilqx
bilqy
bilqz
bilrs
bilrt
bilru
bilrv
bilrw
bilrx
bilry
bilrz
bilst
bilsu
bilsv
bilsw
bilsx
bilsy
bilsz
biltu
biltv
biltw
biltx
bilty
biltz
biluv
biluw
bilux
biluy
biluz
bilvw
bilvx
bilvy
bilvz
bilwx
bilwy
bilwz
bilxy
bilxz
bilyz
bimno
bimnp
bimnq
bimnr
bimns
bimnt
bimnu
bimnv
bimnw
bimnx
bimny
bimnz
bimop
bimoq
bimor
bimos
bimot
bimou
bimov
bimow
bimox
bimoy
bimoz
bimpq
bimpr
bimps
bimpt
bimpu
bimpv
bimpw
bimpx
bimpy
bimpz
bimqr
bimqs
bimqt
bimqu
bimqv
bimqw
bimqx
bimqy
bimqz
bimrs
bimrt
bimru
bimrv
bimrw
bimrx
bimry
bimrz
bimst
bimsu
bimsv
bimsw
bimsx
bimsy
bimsz
bimtu
bimtv
bimtw
bimtx
bimty
bimtz
bimuv
bimuw
bimux
bimuy
bimu

blnry
blnrz
blnst
blnsu
blnsv
blnsw
blnsx
blnsy
blnsz
blntu
blntv
blntw
blntx
blnty
blntz
blnuv
blnuw
blnux
blnuy
blnuz
blnvw
blnvx
blnvy
blnvz
blnwx
blnwy
blnwz
blnxy
blnxz
blnyz
blopq
blopr
blops
blopt
blopu
blopv
blopw
blopx
blopy
blopz
bloqr
bloqs
bloqt
bloqu
bloqv
bloqw
bloqx
bloqy
bloqz
blors
blort
bloru
blorv
blorw
blorx
blory
blorz
blost
blosu
blosv
blosw
blosx
blosy
blosz
blotu
blotv
blotw
blotx
bloty
blotz
blouv
blouw
bloux
blouy
blouz
blovw
blovx
blovy
blovz
blowx
blowy
blowz
bloxy
bloxz
bloyz
blpqr
blpqs
blpqt
blpqu
blpqv
blpqw
blpqx
blpqy
blpqz
blprs
blprt
blpru
blprv
blprw
blprx
blpry
blprz
blpst
blpsu
blpsv
blpsw
blpsx
blpsy
blpsz
blptu
blptv
blptw
blptx
blpty
blptz
blpuv
blpuw
blpux
blpuy
blpuz
blpvw
blpvx
blpvy
blpvz
blpwx
blpwy
blpwz
blpxy
blpxz
blpyz
blqrs
blqrt
blqru
blqrv
blqrw
blqrx
blqry
blqrz
blqst
blqsu
blqsv
blqsw
blqsx
blqsy
blqsz
blqtu
blqtv
blqtw
blqtx
blqty
blqtz
blquv
blquw
blqux
blquy
blquz
blqvw
blqvx
blqvy
blqvz
blqwx
blqwy
blqwz
blqxy
blqxz
blqyz
blrs

cdors
cdort
cdoru
cdorv
cdorw
cdorx
cdory
cdorz
cdost
cdosu
cdosv
cdosw
cdosx
cdosy
cdosz
cdotu
cdotv
cdotw
cdotx
cdoty
cdotz
cdouv
cdouw
cdoux
cdouy
cdouz
cdovw
cdovx
cdovy
cdovz
cdowx
cdowy
cdowz
cdoxy
cdoxz
cdoyz
cdpqr
cdpqs
cdpqt
cdpqu
cdpqv
cdpqw
cdpqx
cdpqy
cdpqz
cdprs
cdprt
cdpru
cdprv
cdprw
cdprx
cdpry
cdprz
cdpst
cdpsu
cdpsv
cdpsw
cdpsx
cdpsy
cdpsz
cdptu
cdptv
cdptw
cdptx
cdpty
cdptz
cdpuv
cdpuw
cdpux
cdpuy
cdpuz
cdpvw
cdpvx
cdpvy
cdpvz
cdpwx
cdpwy
cdpwz
cdpxy
cdpxz
cdpyz
cdqrs
cdqrt
cdqru
cdqrv
cdqrw
cdqrx
cdqry
cdqrz
cdqst
cdqsu
cdqsv
cdqsw
cdqsx
cdqsy
cdqsz
cdqtu
cdqtv
cdqtw
cdqtx
cdqty
cdqtz
cdquv
cdquw
cdqux
cdquy
cdquz
cdqvw
cdqvx
cdqvy
cdqvz
cdqwx
cdqwy
cdqwz
cdqxy
cdqxz
cdqyz
cdrst
cdrsu
cdrsv
cdrsw
cdrsx
cdrsy
cdrsz
cdrtu
cdrtv
cdrtw
cdrtx
cdrty
cdrtz
cdruv
cdruw
cdrux
cdruy
cdruz
cdrvw
cdrvx
cdrvy
cdrvz
cdrwx
cdrwy
cdrwz
cdrxy
cdrxz
cdryz
cdstu
cdstv
cdstw
cdstx
cdsty
cdstz
cdsuv
cdsuw
cdsux
cdsuy
cdsuz
cdsvw
cdsvx
cdsvy
cdsvz
cdswx
cdswy
cdswz
cdsxy
cdsxz
cdsyz
cdtu

cfjkt
cfjku
cfjkv
cfjkw
cfjkx
cfjky
cfjkz
cfjlm
cfjln
cfjlo
cfjlp
cfjlq
cfjlr
cfjls
cfjlt
cfjlu
cfjlv
cfjlw
cfjlx
cfjly
cfjlz
cfjmn
cfjmo
cfjmp
cfjmq
cfjmr
cfjms
cfjmt
cfjmu
cfjmv
cfjmw
cfjmx
cfjmy
cfjmz
cfjno
cfjnp
cfjnq
cfjnr
cfjns
cfjnt
cfjnu
cfjnv
cfjnw
cfjnx
cfjny
cfjnz
cfjop
cfjoq
cfjor
cfjos
cfjot
cfjou
cfjov
cfjow
cfjox
cfjoy
cfjoz
cfjpq
cfjpr
cfjps
cfjpt
cfjpu
cfjpv
cfjpw
cfjpx
cfjpy
cfjpz
cfjqr
cfjqs
cfjqt
cfjqu
cfjqv
cfjqw
cfjqx
cfjqy
cfjqz
cfjrs
cfjrt
cfjru
cfjrv
cfjrw
cfjrx
cfjry
cfjrz
cfjst
cfjsu
cfjsv
cfjsw
cfjsx
cfjsy
cfjsz
cfjtu
cfjtv
cfjtw
cfjtx
cfjty
cfjtz
cfjuv
cfjuw
cfjux
cfjuy
cfjuz
cfjvw
cfjvx
cfjvy
cfjvz
cfjwx
cfjwy
cfjwz
cfjxy
cfjxz
cfjyz
cfklm
cfkln
cfklo
cfklp
cfklq
cfklr
cfkls
cfklt
cfklu
cfklv
cfklw
cfklx
cfkly
cfklz
cfkmn
cfkmo
cfkmp
cfkmq
cfkmr
cfkms
cfkmt
cfkmu
cfkmv
cfkmw
cfkmx
cfkmy
cfkmz
cfkno
cfknp
cfknq
cfknr
cfkns
cfknt
cfknu
cfknv
cfknw
cfknx
cfkny
cfknz
cfkop
cfkoq
cfkor
cfkos
cfkot
cfkou
cfkov
cfkow
cfkox
cfkoy
cfkoz
cfkpq
cfkpr
cfkps
cfkpt
cfkp

chkxz
chkyz
chlmn
chlmo
chlmp
chlmq
chlmr
chlms
chlmt
chlmu
chlmv
chlmw
chlmx
chlmy
chlmz
chlno
chlnp
chlnq
chlnr
chlns
chlnt
chlnu
chlnv
chlnw
chlnx
chlny
chlnz
chlop
chloq
chlor
chlos
chlot
chlou
chlov
chlow
chlox
chloy
chloz
chlpq
chlpr
chlps
chlpt
chlpu
chlpv
chlpw
chlpx
chlpy
chlpz
chlqr
chlqs
chlqt
chlqu
chlqv
chlqw
chlqx
chlqy
chlqz
chlrs
chlrt
chlru
chlrv
chlrw
chlrx
chlry
chlrz
chlst
chlsu
chlsv
chlsw
chlsx
chlsy
chlsz
chltu
chltv
chltw
chltx
chlty
chltz
chluv
chluw
chlux
chluy
chluz
chlvw
chlvx
chlvy
chlvz
chlwx
chlwy
chlwz
chlxy
chlxz
chlyz
chmno
chmnp
chmnq
chmnr
chmns
chmnt
chmnu
chmnv
chmnw
chmnx
chmny
chmnz
chmop
chmoq
chmor
chmos
chmot
chmou
chmov
chmow
chmox
chmoy
chmoz
chmpq
chmpr
chmps
chmpt
chmpu
chmpv
chmpw
chmpx
chmpy
chmpz
chmqr
chmqs
chmqt
chmqu
chmqv
chmqw
chmqx
chmqy
chmqz
chmrs
chmrt
chmru
chmrv
chmrw
chmrx
chmry
chmrz
chmst
chmsu
chmsv
chmsw
chmsx
chmsy
chmsz
chmtu
chmtv
chmtw
chmtx
chmty
chmtz
chmuv
chmuw
chmux
chmuy
chmuz
chmvw
chmvx
chmvy
chmvz
chmwx
chmw

cjorv
cjorw
cjorx
cjory
cjorz
cjost
cjosu
cjosv
cjosw
cjosx
cjosy
cjosz
cjotu
cjotv
cjotw
cjotx
cjoty
cjotz
cjouv
cjouw
cjoux
cjouy
cjouz
cjovw
cjovx
cjovy
cjovz
cjowx
cjowy
cjowz
cjoxy
cjoxz
cjoyz
cjpqr
cjpqs
cjpqt
cjpqu
cjpqv
cjpqw
cjpqx
cjpqy
cjpqz
cjprs
cjprt
cjpru
cjprv
cjprw
cjprx
cjpry
cjprz
cjpst
cjpsu
cjpsv
cjpsw
cjpsx
cjpsy
cjpsz
cjptu
cjptv
cjptw
cjptx
cjpty
cjptz
cjpuv
cjpuw
cjpux
cjpuy
cjpuz
cjpvw
cjpvx
cjpvy
cjpvz
cjpwx
cjpwy
cjpwz
cjpxy
cjpxz
cjpyz
cjqrs
cjqrt
cjqru
cjqrv
cjqrw
cjqrx
cjqry
cjqrz
cjqst
cjqsu
cjqsv
cjqsw
cjqsx
cjqsy
cjqsz
cjqtu
cjqtv
cjqtw
cjqtx
cjqty
cjqtz
cjquv
cjquw
cjqux
cjquy
cjquz
cjqvw
cjqvx
cjqvy
cjqvz
cjqwx
cjqwy
cjqwz
cjqxy
cjqxz
cjqyz
cjrst
cjrsu
cjrsv
cjrsw
cjrsx
cjrsy
cjrsz
cjrtu
cjrtv
cjrtw
cjrtx
cjrty
cjrtz
cjruv
cjruw
cjrux
cjruy
cjruz
cjrvw
cjrvx
cjrvy
cjrvz
cjrwx
cjrwy
cjrwz
cjrxy
cjrxz
cjryz
cjstu
cjstv
cjstw
cjstx
cjsty
cjstz
cjsuv
cjsuw
cjsux
cjsuy
cjsuz
cjsvw
cjsvx
cjsvy
cjsvz
cjswx
cjswy
cjswz
cjsxy
cjsxz
cjsyz
cjtuv
cjtuw
cjtux
cjtu

cqsuy
cqsuz
cqsvw
cqsvx
cqsvy
cqsvz
cqswx
cqswy
cqswz
cqsxy
cqsxz
cqsyz
cqtuv
cqtuw
cqtux
cqtuy
cqtuz
cqtvw
cqtvx
cqtvy
cqtvz
cqtwx
cqtwy
cqtwz
cqtxy
cqtxz
cqtyz
cquvw
cquvx
cquvy
cquvz
cquwx
cquwy
cquwz
cquxy
cquxz
cquyz
cqvwx
cqvwy
cqvwz
cqvxy
cqvxz
cqvyz
cqwxy
cqwxz
cqwyz
cqxyz
crstu
crstv
crstw
crstx
crsty
crstz
crsuv
crsuw
crsux
crsuy
crsuz
crsvw
crsvx
crsvy
crsvz
crswx
crswy
crswz
crsxy
crsxz
crsyz
crtuv
crtuw
crtux
crtuy
crtuz
crtvw
crtvx
crtvy
crtvz
crtwx
crtwy
crtwz
crtxy
crtxz
crtyz
cruvw
cruvx
cruvy
cruvz
cruwx
cruwy
cruwz
cruxy
cruxz
cruyz
crvwx
crvwy
crvwz
crvxy
crvxz
crvyz
crwxy
crwxz
crwyz
crxyz
cstuv
cstuw
cstux
cstuy
cstuz
cstvw
cstvx
cstvy
cstvz
cstwx
cstwy
cstwz
cstxy
cstxz
cstyz
csuvw
csuvx
csuvy
csuvz
csuwx
csuwy
csuwz
csuxy
csuxz
csuyz
csvwx
csvwy
csvwz
csvxy
csvxz
csvyz
cswxy
cswxz
cswyz
csxyz
ctuvw
ctuvx
ctuvy
ctuvz
ctuwx
ctuwy
ctuwz
ctuxy
ctuxz
ctuyz
ctvwx
ctvwy
ctvwz
ctvxy
ctvxz
ctvyz
ctwxy
ctwxz
ctwyz
ctxyz
cuvwx
cuvwy
cuvwz
cuvxy
cuvxz
cuvyz
cuwxy
cuwxz
cuwy

dghiu
dghiv
dghiw
dghix
dghiy
dghiz
dghjk
dghjl
dghjm
dghjn
dghjo
dghjp
dghjq
dghjr
dghjs
dghjt
dghju
dghjv
dghjw
dghjx
dghjy
dghjz
dghkl
dghkm
dghkn
dghko
dghkp
dghkq
dghkr
dghks
dghkt
dghku
dghkv
dghkw
dghkx
dghky
dghkz
dghlm
dghln
dghlo
dghlp
dghlq
dghlr
dghls
dghlt
dghlu
dghlv
dghlw
dghlx
dghly
dghlz
dghmn
dghmo
dghmp
dghmq
dghmr
dghms
dghmt
dghmu
dghmv
dghmw
dghmx
dghmy
dghmz
dghno
dghnp
dghnq
dghnr
dghns
dghnt
dghnu
dghnv
dghnw
dghnx
dghny
dghnz
dghop
dghoq
dghor
dghos
dghot
dghou
dghov
dghow
dghox
dghoy
dghoz
dghpq
dghpr
dghps
dghpt
dghpu
dghpv
dghpw
dghpx
dghpy
dghpz
dghqr
dghqs
dghqt
dghqu
dghqv
dghqw
dghqx
dghqy
dghqz
dghrs
dghrt
dghru
dghrv
dghrw
dghrx
dghry
dghrz
dghst
dghsu
dghsv
dghsw
dghsx
dghsy
dghsz
dghtu
dghtv
dghtw
dghtx
dghty
dghtz
dghuv
dghuw
dghux
dghuy
dghuz
dghvw
dghvx
dghvy
dghvz
dghwx
dghwy
dghwz
dghxy
dghxz
dghyz
dgijk
dgijl
dgijm
dgijn
dgijo
dgijp
dgijq
dgijr
dgijs
dgijt
dgiju
dgijv
dgijw
dgijx
dgijy
dgijz
dgikl
dgikm
dgikn
dgiko
dgikp
dgikq
dgikr
dgiks
dgik

dilmn
dilmo
dilmp
dilmq
dilmr
dilms
dilmt
dilmu
dilmv
dilmw
dilmx
dilmy
dilmz
dilno
dilnp
dilnq
dilnr
dilns
dilnt
dilnu
dilnv
dilnw
dilnx
dilny
dilnz
dilop
diloq
dilor
dilos
dilot
dilou
dilov
dilow
dilox
diloy
diloz
dilpq
dilpr
dilps
dilpt
dilpu
dilpv
dilpw
dilpx
dilpy
dilpz
dilqr
dilqs
dilqt
dilqu
dilqv
dilqw
dilqx
dilqy
dilqz
dilrs
dilrt
dilru
dilrv
dilrw
dilrx
dilry
dilrz
dilst
dilsu
dilsv
dilsw
dilsx
dilsy
dilsz
diltu
diltv
diltw
diltx
dilty
diltz
diluv
diluw
dilux
diluy
diluz
dilvw
dilvx
dilvy
dilvz
dilwx
dilwy
dilwz
dilxy
dilxz
dilyz
dimno
dimnp
dimnq
dimnr
dimns
dimnt
dimnu
dimnv
dimnw
dimnx
dimny
dimnz
dimop
dimoq
dimor
dimos
dimot
dimou
dimov
dimow
dimox
dimoy
dimoz
dimpq
dimpr
dimps
dimpt
dimpu
dimpv
dimpw
dimpx
dimpy
dimpz
dimqr
dimqs
dimqt
dimqu
dimqv
dimqw
dimqx
dimqy
dimqz
dimrs
dimrt
dimru
dimrv
dimrw
dimrx
dimry
dimrz
dimst
dimsu
dimsv
dimsw
dimsx
dimsy
dimsz
dimtu
dimtv
dimtw
dimtx
dimty
dimtz
dimuv
dimuw
dimux
dimuy
dimuz
dimvw
dimvx
dimvy
dimvz
dimwx
dimwy
dimwz
dimx

dmqrs
dmqrt
dmqru
dmqrv
dmqrw
dmqrx
dmqry
dmqrz
dmqst
dmqsu
dmqsv
dmqsw
dmqsx
dmqsy
dmqsz
dmqtu
dmqtv
dmqtw
dmqtx
dmqty
dmqtz
dmquv
dmquw
dmqux
dmquy
dmquz
dmqvw
dmqvx
dmqvy
dmqvz
dmqwx
dmqwy
dmqwz
dmqxy
dmqxz
dmqyz
dmrst
dmrsu
dmrsv
dmrsw
dmrsx
dmrsy
dmrsz
dmrtu
dmrtv
dmrtw
dmrtx
dmrty
dmrtz
dmruv
dmruw
dmrux
dmruy
dmruz
dmrvw
dmrvx
dmrvy
dmrvz
dmrwx
dmrwy
dmrwz
dmrxy
dmrxz
dmryz
dmstu
dmstv
dmstw
dmstx
dmsty
dmstz
dmsuv
dmsuw
dmsux
dmsuy
dmsuz
dmsvw
dmsvx
dmsvy
dmsvz
dmswx
dmswy
dmswz
dmsxy
dmsxz
dmsyz
dmtuv
dmtuw
dmtux
dmtuy
dmtuz
dmtvw
dmtvx
dmtvy
dmtvz
dmtwx
dmtwy
dmtwz
dmtxy
dmtxz
dmtyz
dmuvw
dmuvx
dmuvy
dmuvz
dmuwx
dmuwy
dmuwz
dmuxy
dmuxz
dmuyz
dmvwx
dmvwy
dmvwz
dmvxy
dmvxz
dmvyz
dmwxy
dmwxz
dmwyz
dmxyz
dnopq
dnopr
dnops
dnopt
dnopu
dnopv
dnopw
dnopx
dnopy
dnopz
dnoqr
dnoqs
dnoqt
dnoqu
dnoqv
dnoqw
dnoqx
dnoqy
dnoqz
dnors
dnort
dnoru
dnorv
dnorw
dnorx
dnory
dnorz
dnost
dnosu
dnosv
dnosw
dnosx
dnosy
dnosz
dnotu
dnotv
dnotw
dnotx
dnoty
dnotz
dnouv
dnouw
dnoux
dnouy
dnouz
dnovw
dnov

efktu
efktv
efktw
efktx
efkty
efktz
efkuv
efkuw
efkux
efkuy
efkuz
efkvw
efkvx
efkvy
efkvz
efkwx
efkwy
efkwz
efkxy
efkxz
efkyz
eflmn
eflmo
eflmp
eflmq
eflmr
eflms
eflmt
eflmu
eflmv
eflmw
eflmx
eflmy
eflmz
eflno
eflnp
eflnq
eflnr
eflns
eflnt
eflnu
eflnv
eflnw
eflnx
eflny
eflnz
eflop
efloq
eflor
eflos
eflot
eflou
eflov
eflow
eflox
efloy
efloz
eflpq
eflpr
eflps
eflpt
eflpu
eflpv
eflpw
eflpx
eflpy
eflpz
eflqr
eflqs
eflqt
eflqu
eflqv
eflqw
eflqx
eflqy
eflqz
eflrs
eflrt
eflru
eflrv
eflrw
eflrx
eflry
eflrz
eflst
eflsu
eflsv
eflsw
eflsx
eflsy
eflsz
efltu
efltv
efltw
efltx
eflty
efltz
efluv
efluw
eflux
efluy
efluz
eflvw
eflvx
eflvy
eflvz
eflwx
eflwy
eflwz
eflxy
eflxz
eflyz
efmno
efmnp
efmnq
efmnr
efmns
efmnt
efmnu
efmnv
efmnw
efmnx
efmny
efmnz
efmop
efmoq
efmor
efmos
efmot
efmou
efmov
efmow
efmox
efmoy
efmoz
efmpq
efmpr
efmps
efmpt
efmpu
efmpv
efmpw
efmpx
efmpy
efmpz
efmqr
efmqs
efmqt
efmqu
efmqv
efmqw
efmqx
efmqy
efmqz
efmrs
efmrt
efmru
efmrv
efmrw
efmrx
efmry
efmrz
efmst
efmsu
efmsv
efmsw
efms

ehnqu
ehnqv
ehnqw
ehnqx
ehnqy
ehnqz
ehnrs
ehnrt
ehnru
ehnrv
ehnrw
ehnrx
ehnry
ehnrz
ehnst
ehnsu
ehnsv
ehnsw
ehnsx
ehnsy
ehnsz
ehntu
ehntv
ehntw
ehntx
ehnty
ehntz
ehnuv
ehnuw
ehnux
ehnuy
ehnuz
ehnvw
ehnvx
ehnvy
ehnvz
ehnwx
ehnwy
ehnwz
ehnxy
ehnxz
ehnyz
ehopq
ehopr
ehops
ehopt
ehopu
ehopv
ehopw
ehopx
ehopy
ehopz
ehoqr
ehoqs
ehoqt
ehoqu
ehoqv
ehoqw
ehoqx
ehoqy
ehoqz
ehors
ehort
ehoru
ehorv
ehorw
ehorx
ehory
ehorz
ehost
ehosu
ehosv
ehosw
ehosx
ehosy
ehosz
ehotu
ehotv
ehotw
ehotx
ehoty
ehotz
ehouv
ehouw
ehoux
ehouy
ehouz
ehovw
ehovx
ehovy
ehovz
ehowx
ehowy
ehowz
ehoxy
ehoxz
ehoyz
ehpqr
ehpqs
ehpqt
ehpqu
ehpqv
ehpqw
ehpqx
ehpqy
ehpqz
ehprs
ehprt
ehpru
ehprv
ehprw
ehprx
ehpry
ehprz
ehpst
ehpsu
ehpsv
ehpsw
ehpsx
ehpsy
ehpsz
ehptu
ehptv
ehptw
ehptx
ehpty
ehptz
ehpuv
ehpuw
ehpux
ehpuy
ehpuz
ehpvw
ehpvx
ehpvy
ehpvz
ehpwx
ehpwy
ehpwz
ehpxy
ehpxz
ehpyz
ehqrs
ehqrt
ehqru
ehqrv
ehqrw
ehqrx
ehqry
ehqrz
ehqst
ehqsu
ehqsv
ehqsw
ehqsx
ehqsy
ehqsz
ehqtu
ehqtv
ehqtw
ehqtx
ehqty
ehqtz
ehquv
ehquw
ehqux
ehqu

emqsx
emqsy
emqsz
emqtu
emqtv
emqtw
emqtx
emqty
emqtz
emquv
emquw
emqux
emquy
emquz
emqvw
emqvx
emqvy
emqvz
emqwx
emqwy
emqwz
emqxy
emqxz
emqyz
emrst
emrsu
emrsv
emrsw
emrsx
emrsy
emrsz
emrtu
emrtv
emrtw
emrtx
emrty
emrtz
emruv
emruw
emrux
emruy
emruz
emrvw
emrvx
emrvy
emrvz
emrwx
emrwy
emrwz
emrxy
emrxz
emryz
emstu
emstv
emstw
emstx
emsty
emstz
emsuv
emsuw
emsux
emsuy
emsuz
emsvw
emsvx
emsvy
emsvz
emswx
emswy
emswz
emsxy
emsxz
emsyz
emtuv
emtuw
emtux
emtuy
emtuz
emtvw
emtvx
emtvy
emtvz
emtwx
emtwy
emtwz
emtxy
emtxz
emtyz
emuvw
emuvx
emuvy
emuvz
emuwx
emuwy
emuwz
emuxy
emuxz
emuyz
emvwx
emvwy
emvwz
emvxy
emvxz
emvyz
emwxy
emwxz
emwyz
emxyz
enopq
enopr
enops
enopt
enopu
enopv
enopw
enopx
enopy
enopz
enoqr
enoqs
enoqt
enoqu
enoqv
enoqw
enoqx
enoqy
enoqz
enors
enort
enoru
enorv
enorw
enorx
enory
enorz
enost
enosu
enosv
enosw
enosx
enosy
enosz
enotu
enotv
enotw
enotx
enoty
enotz
enouv
enouw
enoux
enouy
enouz
enovw
enovx
enovy
enovz
enowx
enowy
enowz
enoxy
enoxz
enoyz
enpqr
enpqs
enpqt
enpq

fgmwx
fgmwy
fgmwz
fgmxy
fgmxz
fgmyz
fgnop
fgnoq
fgnor
fgnos
fgnot
fgnou
fgnov
fgnow
fgnox
fgnoy
fgnoz
fgnpq
fgnpr
fgnps
fgnpt
fgnpu
fgnpv
fgnpw
fgnpx
fgnpy
fgnpz
fgnqr
fgnqs
fgnqt
fgnqu
fgnqv
fgnqw
fgnqx
fgnqy
fgnqz
fgnrs
fgnrt
fgnru
fgnrv
fgnrw
fgnrx
fgnry
fgnrz
fgnst
fgnsu
fgnsv
fgnsw
fgnsx
fgnsy
fgnsz
fgntu
fgntv
fgntw
fgntx
fgnty
fgntz
fgnuv
fgnuw
fgnux
fgnuy
fgnuz
fgnvw
fgnvx
fgnvy
fgnvz
fgnwx
fgnwy
fgnwz
fgnxy
fgnxz
fgnyz
fgopq
fgopr
fgops
fgopt
fgopu
fgopv
fgopw
fgopx
fgopy
fgopz
fgoqr
fgoqs
fgoqt
fgoqu
fgoqv
fgoqw
fgoqx
fgoqy
fgoqz
fgors
fgort
fgoru
fgorv
fgorw
fgorx
fgory
fgorz
fgost
fgosu
fgosv
fgosw
fgosx
fgosy
fgosz
fgotu
fgotv
fgotw
fgotx
fgoty
fgotz
fgouv
fgouw
fgoux
fgouy
fgouz
fgovw
fgovx
fgovy
fgovz
fgowx
fgowy
fgowz
fgoxy
fgoxz
fgoyz
fgpqr
fgpqs
fgpqt
fgpqu
fgpqv
fgpqw
fgpqx
fgpqy
fgpqz
fgprs
fgprt
fgpru
fgprv
fgprw
fgprx
fgpry
fgprz
fgpst
fgpsu
fgpsv
fgpsw
fgpsx
fgpsy
fgpsz
fgptu
fgptv
fgptw
fgptx
fgpty
fgptz
fgpuv
fgpuw
fgpux
fgpuy
fgpuz
fgpvw
fgpvx
fgpvy
fgpvz
fgpw

fjmos
fjmot
fjmou
fjmov
fjmow
fjmox
fjmoy
fjmoz
fjmpq
fjmpr
fjmps
fjmpt
fjmpu
fjmpv
fjmpw
fjmpx
fjmpy
fjmpz
fjmqr
fjmqs
fjmqt
fjmqu
fjmqv
fjmqw
fjmqx
fjmqy
fjmqz
fjmrs
fjmrt
fjmru
fjmrv
fjmrw
fjmrx
fjmry
fjmrz
fjmst
fjmsu
fjmsv
fjmsw
fjmsx
fjmsy
fjmsz
fjmtu
fjmtv
fjmtw
fjmtx
fjmty
fjmtz
fjmuv
fjmuw
fjmux
fjmuy
fjmuz
fjmvw
fjmvx
fjmvy
fjmvz
fjmwx
fjmwy
fjmwz
fjmxy
fjmxz
fjmyz
fjnop
fjnoq
fjnor
fjnos
fjnot
fjnou
fjnov
fjnow
fjnox
fjnoy
fjnoz
fjnpq
fjnpr
fjnps
fjnpt
fjnpu
fjnpv
fjnpw
fjnpx
fjnpy
fjnpz
fjnqr
fjnqs
fjnqt
fjnqu
fjnqv
fjnqw
fjnqx
fjnqy
fjnqz
fjnrs
fjnrt
fjnru
fjnrv
fjnrw
fjnrx
fjnry
fjnrz
fjnst
fjnsu
fjnsv
fjnsw
fjnsx
fjnsy
fjnsz
fjntu
fjntv
fjntw
fjntx
fjnty
fjntz
fjnuv
fjnuw
fjnux
fjnuy
fjnuz
fjnvw
fjnvx
fjnvy
fjnvz
fjnwx
fjnwy
fjnwz
fjnxy
fjnxz
fjnyz
fjopq
fjopr
fjops
fjopt
fjopu
fjopv
fjopw
fjopx
fjopy
fjopz
fjoqr
fjoqs
fjoqt
fjoqu
fjoqv
fjoqw
fjoqx
fjoqy
fjoqz
fjors
fjort
fjoru
fjorv
fjorw
fjorx
fjory
fjorz
fjost
fjosu
fjosv
fjosw
fjosx
fjosy
fjosz
fjotu
fjotv
fjotw
fjot

fqsvy
fqsvz
fqswx
fqswy
fqswz
fqsxy
fqsxz
fqsyz
fqtuv
fqtuw
fqtux
fqtuy
fqtuz
fqtvw
fqtvx
fqtvy
fqtvz
fqtwx
fqtwy
fqtwz
fqtxy
fqtxz
fqtyz
fquvw
fquvx
fquvy
fquvz
fquwx
fquwy
fquwz
fquxy
fquxz
fquyz
fqvwx
fqvwy
fqvwz
fqvxy
fqvxz
fqvyz
fqwxy
fqwxz
fqwyz
fqxyz
frstu
frstv
frstw
frstx
frsty
frstz
frsuv
frsuw
frsux
frsuy
frsuz
frsvw
frsvx
frsvy
frsvz
frswx
frswy
frswz
frsxy
frsxz
frsyz
frtuv
frtuw
frtux
frtuy
frtuz
frtvw
frtvx
frtvy
frtvz
frtwx
frtwy
frtwz
frtxy
frtxz
frtyz
fruvw
fruvx
fruvy
fruvz
fruwx
fruwy
fruwz
fruxy
fruxz
fruyz
frvwx
frvwy
frvwz
frvxy
frvxz
frvyz
frwxy
frwxz
frwyz
frxyz
fstuv
fstuw
fstux
fstuy
fstuz
fstvw
fstvx
fstvy
fstvz
fstwx
fstwy
fstwz
fstxy
fstxz
fstyz
fsuvw
fsuvx
fsuvy
fsuvz
fsuwx
fsuwy
fsuwz
fsuxy
fsuxz
fsuyz
fsvwx
fsvwy
fsvwz
fsvxy
fsvxz
fsvyz
fswxy
fswxz
fswyz
fsxyz
ftuvw
ftuvx
ftuvy
ftuvz
ftuwx
ftuwy
ftuwz
ftuxy
ftuxz
ftuyz
ftvwx
ftvwy
ftvwz
ftvxy
ftvxz
ftvyz
ftwxy
ftwxz
ftwyz
ftxyz
fuvwx
fuvwy
fuvwz
fuvxy
fuvxz
fuvyz
fuwxy
fuwxz
fuwyz
fuxyz
fvwxy
fvwxz
fvwy

gjkqw
gjkqx
gjkqy
gjkqz
gjkrs
gjkrt
gjkru
gjkrv
gjkrw
gjkrx
gjkry
gjkrz
gjkst
gjksu
gjksv
gjksw
gjksx
gjksy
gjksz
gjktu
gjktv
gjktw
gjktx
gjkty
gjktz
gjkuv
gjkuw
gjkux
gjkuy
gjkuz
gjkvw
gjkvx
gjkvy
gjkvz
gjkwx
gjkwy
gjkwz
gjkxy
gjkxz
gjkyz
gjlmn
gjlmo
gjlmp
gjlmq
gjlmr
gjlms
gjlmt
gjlmu
gjlmv
gjlmw
gjlmx
gjlmy
gjlmz
gjlno
gjlnp
gjlnq
gjlnr
gjlns
gjlnt
gjlnu
gjlnv
gjlnw
gjlnx
gjlny
gjlnz
gjlop
gjloq
gjlor
gjlos
gjlot
gjlou
gjlov
gjlow
gjlox
gjloy
gjloz
gjlpq
gjlpr
gjlps
gjlpt
gjlpu
gjlpv
gjlpw
gjlpx
gjlpy
gjlpz
gjlqr
gjlqs
gjlqt
gjlqu
gjlqv
gjlqw
gjlqx
gjlqy
gjlqz
gjlrs
gjlrt
gjlru
gjlrv
gjlrw
gjlrx
gjlry
gjlrz
gjlst
gjlsu
gjlsv
gjlsw
gjlsx
gjlsy
gjlsz
gjltu
gjltv
gjltw
gjltx
gjlty
gjltz
gjluv
gjluw
gjlux
gjluy
gjluz
gjlvw
gjlvx
gjlvy
gjlvz
gjlwx
gjlwy
gjlwz
gjlxy
gjlxz
gjlyz
gjmno
gjmnp
gjmnq
gjmnr
gjmns
gjmnt
gjmnu
gjmnv
gjmnw
gjmnx
gjmny
gjmnz
gjmop
gjmoq
gjmor
gjmos
gjmot
gjmou
gjmov
gjmow
gjmox
gjmoy
gjmoz
gjmpq
gjmpr
gjmps
gjmpt
gjmpu
gjmpv
gjmpw
gjmpx
gjmpy
gjmpz
gjmqr
gjmqs
gjmq

hjmyz
hjnop
hjnoq
hjnor
hjnos
hjnot
hjnou
hjnov
hjnow
hjnox
hjnoy
hjnoz
hjnpq
hjnpr
hjnps
hjnpt
hjnpu
hjnpv
hjnpw
hjnpx
hjnpy
hjnpz
hjnqr
hjnqs
hjnqt
hjnqu
hjnqv
hjnqw
hjnqx
hjnqy
hjnqz
hjnrs
hjnrt
hjnru
hjnrv
hjnrw
hjnrx
hjnry
hjnrz
hjnst
hjnsu
hjnsv
hjnsw
hjnsx
hjnsy
hjnsz
hjntu
hjntv
hjntw
hjntx
hjnty
hjntz
hjnuv
hjnuw
hjnux
hjnuy
hjnuz
hjnvw
hjnvx
hjnvy
hjnvz
hjnwx
hjnwy
hjnwz
hjnxy
hjnxz
hjnyz
hjopq
hjopr
hjops
hjopt
hjopu
hjopv
hjopw
hjopx
hjopy
hjopz
hjoqr
hjoqs
hjoqt
hjoqu
hjoqv
hjoqw
hjoqx
hjoqy
hjoqz
hjors
hjort
hjoru
hjorv
hjorw
hjorx
hjory
hjorz
hjost
hjosu
hjosv
hjosw
hjosx
hjosy
hjosz
hjotu
hjotv
hjotw
hjotx
hjoty
hjotz
hjouv
hjouw
hjoux
hjouy
hjouz
hjovw
hjovx
hjovy
hjovz
hjowx
hjowy
hjowz
hjoxy
hjoxz
hjoyz
hjpqr
hjpqs
hjpqt
hjpqu
hjpqv
hjpqw
hjpqx
hjpqy
hjpqz
hjprs
hjprt
hjpru
hjprv
hjprw
hjprx
hjpry
hjprz
hjpst
hjpsu
hjpsv
hjpsw
hjpsx
hjpsy
hjpsz
hjptu
hjptv
hjptw
hjptx
hjpty
hjptz
hjpuv
hjpuw
hjpux
hjpuy
hjpuz
hjpvw
hjpvx
hjpvy
hjpvz
hjpwx
hjpwy
hjpwz
hjpxy
hjpxz
hjpy

hnvxz
hnvyz
hnwxy
hnwxz
hnwyz
hnxyz
hopqr
hopqs
hopqt
hopqu
hopqv
hopqw
hopqx
hopqy
hopqz
hoprs
hoprt
hopru
hoprv
hoprw
hoprx
hopry
hoprz
hopst
hopsu
hopsv
hopsw
hopsx
hopsy
hopsz
hoptu
hoptv
hoptw
hoptx
hopty
hoptz
hopuv
hopuw
hopux
hopuy
hopuz
hopvw
hopvx
hopvy
hopvz
hopwx
hopwy
hopwz
hopxy
hopxz
hopyz
hoqrs
hoqrt
hoqru
hoqrv
hoqrw
hoqrx
hoqry
hoqrz
hoqst
hoqsu
hoqsv
hoqsw
hoqsx
hoqsy
hoqsz
hoqtu
hoqtv
hoqtw
hoqtx
hoqty
hoqtz
hoquv
hoquw
hoqux
hoquy
hoquz
hoqvw
hoqvx
hoqvy
hoqvz
hoqwx
hoqwy
hoqwz
hoqxy
hoqxz
hoqyz
horst
horsu
horsv
horsw
horsx
horsy
horsz
hortu
hortv
hortw
hortx
horty
hortz
horuv
horuw
horux
horuy
horuz
horvw
horvx
horvy
horvz
horwx
horwy
horwz
horxy
horxz
horyz
hostu
hostv
hostw
hostx
hosty
hostz
hosuv
hosuw
hosux
hosuy
hosuz
hosvw
hosvx
hosvy
hosvz
hoswx
hoswy
hoswz
hosxy
hosxz
hosyz
hotuv
hotuw
hotux
hotuy
hotuz
hotvw
hotvx
hotvy
hotvz
hotwx
hotwy
hotwz
hotxy
hotxz
hotyz
houvw
houvx
houvy
houvz
houwx
houwy
houwz
houxy
houxz
houyz
hovwx
hovwy
hovwz
hovxy
hovxz
hovy

imnps
imnpt
imnpu
imnpv
imnpw
imnpx
imnpy
imnpz
imnqr
imnqs
imnqt
imnqu
imnqv
imnqw
imnqx
imnqy
imnqz
imnrs
imnrt
imnru
imnrv
imnrw
imnrx
imnry
imnrz
imnst
imnsu
imnsv
imnsw
imnsx
imnsy
imnsz
imntu
imntv
imntw
imntx
imnty
imntz
imnuv
imnuw
imnux
imnuy
imnuz
imnvw
imnvx
imnvy
imnvz
imnwx
imnwy
imnwz
imnxy
imnxz
imnyz
imopq
imopr
imops
imopt
imopu
imopv
imopw
imopx
imopy
imopz
imoqr
imoqs
imoqt
imoqu
imoqv
imoqw
imoqx
imoqy
imoqz
imors
imort
imoru
imorv
imorw
imorx
imory
imorz
imost
imosu
imosv
imosw
imosx
imosy
imosz
imotu
imotv
imotw
imotx
imoty
imotz
imouv
imouw
imoux
imouy
imouz
imovw
imovx
imovy
imovz
imowx
imowy
imowz
imoxy
imoxz
imoyz
impqr
impqs
impqt
impqu
impqv
impqw
impqx
impqy
impqz
imprs
imprt
impru
imprv
imprw
imprx
impry
imprz
impst
impsu
impsv
impsw
impsx
impsy
impsz
imptu
imptv
imptw
imptx
impty
imptz
impuv
impuw
impux
impuy
impuz
impvw
impvx
impvy
impvz
impwx
impwy
impwz
impxy
impxz
impyz
imqrs
imqrt
imqru
imqrv
imqrw
imqrx
imqry
imqrz
imqst
imqsu
imqsv
imqsw
imqsx
imqs

jmqvx
jmqvy
jmqvz
jmqwx
jmqwy
jmqwz
jmqxy
jmqxz
jmqyz
jmrst
jmrsu
jmrsv
jmrsw
jmrsx
jmrsy
jmrsz
jmrtu
jmrtv
jmrtw
jmrtx
jmrty
jmrtz
jmruv
jmruw
jmrux
jmruy
jmruz
jmrvw
jmrvx
jmrvy
jmrvz
jmrwx
jmrwy
jmrwz
jmrxy
jmrxz
jmryz
jmstu
jmstv
jmstw
jmstx
jmsty
jmstz
jmsuv
jmsuw
jmsux
jmsuy
jmsuz
jmsvw
jmsvx
jmsvy
jmsvz
jmswx
jmswy
jmswz
jmsxy
jmsxz
jmsyz
jmtuv
jmtuw
jmtux
jmtuy
jmtuz
jmtvw
jmtvx
jmtvy
jmtvz
jmtwx
jmtwy
jmtwz
jmtxy
jmtxz
jmtyz
jmuvw
jmuvx
jmuvy
jmuvz
jmuwx
jmuwy
jmuwz
jmuxy
jmuxz
jmuyz
jmvwx
jmvwy
jmvwz
jmvxy
jmvxz
jmvyz
jmwxy
jmwxz
jmwyz
jmxyz
jnopq
jnopr
jnops
jnopt
jnopu
jnopv
jnopw
jnopx
jnopy
jnopz
jnoqr
jnoqs
jnoqt
jnoqu
jnoqv
jnoqw
jnoqx
jnoqy
jnoqz
jnors
jnort
jnoru
jnorv
jnorw
jnorx
jnory
jnorz
jnost
jnosu
jnosv
jnosw
jnosx
jnosy
jnosz
jnotu
jnotv
jnotw
jnotx
jnoty
jnotz
jnouv
jnouw
jnoux
jnouy
jnouz
jnovw
jnovx
jnovy
jnovz
jnowx
jnowy
jnowz
jnoxy
jnoxz
jnoyz
jnpqr
jnpqs
jnpqt
jnpqu
jnpqv
jnpqw
jnpqx
jnpqy
jnpqz
jnprs
jnprt
jnpru
jnprv
jnprw
jnprx
jnpry
jnprz
jnpst
jnps

kqsux
kqsuy
kqsuz
kqsvw
kqsvx
kqsvy
kqsvz
kqswx
kqswy
kqswz
kqsxy
kqsxz
kqsyz
kqtuv
kqtuw
kqtux
kqtuy
kqtuz
kqtvw
kqtvx
kqtvy
kqtvz
kqtwx
kqtwy
kqtwz
kqtxy
kqtxz
kqtyz
kquvw
kquvx
kquvy
kquvz
kquwx
kquwy
kquwz
kquxy
kquxz
kquyz
kqvwx
kqvwy
kqvwz
kqvxy
kqvxz
kqvyz
kqwxy
kqwxz
kqwyz
kqxyz
krstu
krstv
krstw
krstx
krsty
krstz
krsuv
krsuw
krsux
krsuy
krsuz
krsvw
krsvx
krsvy
krsvz
krswx
krswy
krswz
krsxy
krsxz
krsyz
krtuv
krtuw
krtux
krtuy
krtuz
krtvw
krtvx
krtvy
krtvz
krtwx
krtwy
krtwz
krtxy
krtxz
krtyz
kruvw
kruvx
kruvy
kruvz
kruwx
kruwy
kruwz
kruxy
kruxz
kruyz
krvwx
krvwy
krvwz
krvxy
krvxz
krvyz
krwxy
krwxz
krwyz
krxyz
kstuv
kstuw
kstux
kstuy
kstuz
kstvw
kstvx
kstvy
kstvz
kstwx
kstwy
kstwz
kstxy
kstxz
kstyz
ksuvw
ksuvx
ksuvy
ksuvz
ksuwx
ksuwy
ksuwz
ksuxy
ksuxz
ksuyz
ksvwx
ksvwy
ksvwz
ksvxy
ksvxz
ksvyz
kswxy
kswxz
kswyz
ksxyz
ktuvw
ktuvx
ktuvy
ktuvz
ktuwx
ktuwy
ktuwz
ktuxy
ktuxz
ktuyz
ktvwx
ktvwy
ktvwz
ktvxy
ktvxz
ktvyz
ktwxy
ktwxz
ktwyz
ktxyz
kuvwx
kuvwy
kuvwz
kuvxy
kuvxz
kuvyz
kuwxy
kuwx

nostv
nostw
nostx
nosty
nostz
nosuv
nosuw
nosux
nosuy
nosuz
nosvw
nosvx
nosvy
nosvz
noswx
noswy
noswz
nosxy
nosxz
nosyz
notuv
notuw
notux
notuy
notuz
notvw
notvx
notvy
notvz
notwx
notwy
notwz
notxy
notxz
notyz
nouvw
nouvx
nouvy
nouvz
nouwx
nouwy
nouwz
nouxy
nouxz
nouyz
novwx
novwy
novwz
novxy
novxz
novyz
nowxy
nowxz
nowyz
noxyz
npqrs
npqrt
npqru
npqrv
npqrw
npqrx
npqry
npqrz
npqst
npqsu
npqsv
npqsw
npqsx
npqsy
npqsz
npqtu
npqtv
npqtw
npqtx
npqty
npqtz
npquv
npquw
npqux
npquy
npquz
npqvw
npqvx
npqvy
npqvz
npqwx
npqwy
npqwz
npqxy
npqxz
npqyz
nprst
nprsu
nprsv
nprsw
nprsx
nprsy
nprsz
nprtu
nprtv
nprtw
nprtx
nprty
nprtz
npruv
npruw
nprux
npruy
npruz
nprvw
nprvx
nprvy
nprvz
nprwx
nprwy
nprwz
nprxy
nprxz
npryz
npstu
npstv
npstw
npstx
npsty
npstz
npsuv
npsuw
npsux
npsuy
npsuz
npsvw
npsvx
npsvy
npsvz
npswx
npswy
npswz
npsxy
npsxz
npsyz
nptuv
nptuw
nptux
nptuy
nptuz
nptvw
nptvx
nptvy
nptvz
nptwx
nptwy
nptwz
nptxy
nptxz
nptyz
npuvw
npuvx
npuvy
npuvz
npuwx
npuwy
npuwz
npuxy
npuxz
npuyz
npvwx
npvw

*__There are recursive functions that generate the same combinations - cf. [here](http://amjith.blogspot.com/2011/10/picking-items-from-list-of-recursion.html "recursive function to generate combinations"); but these use `lists`, which have not yet been covered in "Think Python 2e".  Furthermore, recursive functions tend to be slow, which would be an issue in this case.__*

In [17]:
# from http://amjith.blogspot.com/2011/10/picking-items-from-list-of-recursion.html

def combo(w, l):
    lst = []
    for i in range(len(w)):
        if l == 1:
            lst.append(w[i])
        for c in combo(w[i+1:], l-1):
            lst.append(w[i] + c)
    return lst



In [18]:
combo_list = combo(alphabet, 5)

*__I tested all the combinations on a fairly fast machine (Core i7 processor, 8GB RAM), but the following code was very slow and took nearly two hours to run.__*

```
# Tallying and printing words in words.txt without every five letter combination

fin = open('words.txt')

# Tallying number of words in words.txt

tally = 0
for line in fin:
    tally += 1
    
# initializing the tally and setting the least_freq_combo

least_freq_tally = 0
least_freq_combo = ""

# creating all possible combinations 

n = 0
p1 = n
while p1 < len(alphabet):
    p2 = p1 + 1
    while p2 < len(alphabet):
        p3 = p2 + 1
        while p3 < len(alphabet):
            p4 = p3 + 1
            while p4 < len(alphabet):
                p5 = p4 + 1
                while p5 < len(alphabet):
                    forbidden = (alphabet[p1] + alphabet[p2] + alphabet[p3] +
                                 alphabet[p4] + alphabet[p5])
                    avoids_tally = tally
                    fin = open('words.txt')
                    
                    # testing the new combination
                    
                    for line in fin:
                        if avoids(forbidden, line.strip()):
                            avoids_tally -=1
                    
                    # updating the tally and least_freq_combo
                    
                    if avoids_tally > least_freq_tally:
                        least_freq_tally = avoids_tally
                        least_freq_combo = forbidden
                    
                    p5 += 1
                p4 += 1
            p3 += 1
        p2 += 1
    p1 += 1
    
print(least_freq_combo)
print(least_freq_tally)
```

*__After about two hours, I got my result: `jqwxz` excluded the fewest words: 17384, or about 600 fewer that just using the five least common letters in English:__*

In [19]:
fin = open('words.txt')
forbidden = "jqwxz"
avoids_tally = tally
for line in fin:
    if avoids(forbidden, line.strip()):
        avoids_tally -= 1
    
avoids_tally

17384

*__For the sake of comparison I tried to run through the different combinations once more, this time using those produced by the recursive algorithm.  The differences in time to complete were negligible (85 mins. vs. 89 mins.), and the two programs produced the same result.__* 

```
# Tallying and printing words in words.txt without every five letter combination

fin = open('words.txt')

# Tallying number of words in words.txt

tally = 0
for line in fin:
    tally += 1
    
# initializing the tally and setting the least_freq_combo

least_freq_tally = tally
least_freq_combo = ""

for c in combo_list:
    avoids_tally = tally
    fin = open('words.txt')
    for line in fin:
        if avoids(c, line.strip()):
            avoids_tally -=1

    if avoids_tally < least_freq_tally:
        least_freq_tally = avoids_tally
        least_freq_combo = c
```

```
print(least_freq_combo)
print(least_freq_tally)

jqwxz
17384
```

#### Exercise 4  

*Write a function named `uses_only` that takes a word and a string of letters, and that returns `True` if the word contains only letters in the list. Can you make a sentence using only the letters `acefhlo`? Other than “Hoe alfalfa”?*

*__Here are three ways we could write the function.  The first uses the index location to traverse `word`:__*

In [20]:
def uses_only(word, string):
    """
    Returns True if word contains only
    characters also found in string
    """
    i = 0
    while i < len(word):
        if word[i] in string:
            i += 1
        else:
            return False
    return True
        

*__The second uses a rather inelegant combination of `in` and `pass` to eliminate words with characters not in the string:__*

In [21]:
def uses_only(word, string):
    """
    Returns True if word contains only
    characters also found in string
    """
    i = 0
    for char in word:
        if char in string:
            pass
        else:
            return False
    return True

*__The third is more concise, but uses `not`, which I believe has not yet been introduced:__*

In [22]:
def uses_only(word, string):
    """
    Returns True if word contains only
    characters also found in string
    """
    i = 0
    for char in word:
        if char not in string:
            return False
    return True

In [23]:
uses_only('aahed', 'acefhlo')

False

In [24]:
fin = open('words.txt')
for line in fin:
    word = line.strip()
    if uses_only(word, 'acefhlo'):
        print(word)

aa
aah
aal
ace
ache
achoo
ae
aff
ah
aha
ahchoo
ala
alae
alcohol
ale
alec
alee
alef
alfa
alfalfa
all
allele
allheal
aloe
aloha
aloof
cacao
cache
caeca
caecal
cafe
caleche
calf
call
calla
ceca
cecal
cee
cell
cella
cellae
cello
chafe
chaff
chalah
chaleh
challah
chef
chela
chelae
cholla
clach
clef
cloaca
cloacae
cloacal
cloche
coach
coal
coala
coalhole
coca
coccal
cochlea
cochleae
coco
cocoa
coff
coffee
coffle
coho
col
cola
cole
coo
cooch
cooee
coof
cool
each
eche
echo
ecole
eel
ef
eff
efface
eh
el
elf
ell
fa
face
faecal
fall
fallal
falloff
feal
fecal
fee
feel
fell
fella
fellah
felloe
feoff
feoffee
flea
fleche
flee
fleece
fleech
floc
floe
foal
focal
foe
foh
fool
ha
haaf
hae
hah
halala
halalah
hale
half
hall
hallah
hallel
hallo
halloa
halloo
halo
haole
he
heal
heel
hell
hellhole
hello
ho
hoe
hole
holla
hollo
holloa
holloo
hooch
hoof
la
lac
lace
lall
lea
leach
leaf
leal
lech
lee
leech
lo
loach
loaf
loca
local
locale
loch
loco
locofoco
loll
loo
loof
loofa
loofah
oaf
oca
oe
of
off
offal
oh
oho

*__I'm not terribly interested in the second challenge of this exercise, but just for shits & giggles: "each cacao leaf fell, achoo!"__*

#### Exercise 5  

*Write a function named `uses_all` that takes a word and a string of required letters, and that returns `True` if the word uses all the required letters at least once. How many words are there that use all the vowels `aeiou`? How about `aeiouy`?*

*__As with `uses_only`, there are three ways we could write the function.  The first way:__*

In [25]:
def uses_all(word, string):
    """
    Returns True if word uses all
    characters in string
    """
    i = 0
    while i < len(string):
        if string[i] in word:
            i += 1
        else:
            return False
    return True
        

*__The second:__*

In [26]:
def uses_all(word, string):
    """
    Returns True if word uses all
    characters in string
    """
    i = 0
    for char in string:
        if char in word:
            pass
        else:
            return False
    return True

*__The third:__*

In [27]:
def uses_all(word, string):
    """
    Returns True if word uses all
    characters in string
    """
    i = 0
    for char in string:
        if char not in word:
            return False
    return True

In [28]:
uses_all('eulogia', 'aeiou')

True

In [29]:
uses_all('eulogy', 'aeiou')

False

In [30]:
# Words that use all vowels at least once:

fin = open('words.txt')
for line in fin:
    word = line.strip()
    if uses_all(word, 'aeiou'):
        print(word)

aboideau
aboideaus
aboideaux
aboiteau
aboiteaus
aboiteaux
abstemious
abstemiously
accentuation
accentuations
accountabilities
accountancies
accoutering
adulteration
adulterations
adventitious
adventitiously
adventitiousness
adventitiousnesses
aerobium
aeronautic
aeronautical
aeronautically
aeronautics
agouties
ambidextrous
ambidextrously
antibourgeois
anticonsumer
antievolution
antievolutionary
antihomosexual
antireligious
antirevolutionary
antisubversion
antituberculosis
antiunemployment
armouries
arsenious
assiduousness
assiduousnesses
atrociousness
atrociousnesses
attenuation
attenuations
auctioned
auctioneer
auctioneers
auditioned
auditories
augmentation
augmentations
aureoling
authentication
authentications
authoritative
authoritatively
authorities
authorize
authorized
authorizes
autobiographer
autobiographers
autobiographies
autocracies
autogamies
autogenies
automobile
automobiles
automotive
autonomies
autopsied
autopsies
autotomies
autotypies
avoidupoises
beautification
beautifi

In [31]:
# Words that use all vowels and 'y' at least once:

fin = open('words.txt')
for line in fin:
    word = line.strip()
    if uses_all(word, 'aeiouy'):
        print(word)

abstemiously
adventitiously
aeronautically
ambidextrously
antievolutionary
antirevolutionary
antiunemployment
authoritatively
autotypies
buoyancies
counterinflationary
evolutionary
extracommunity
facetiously
genitourinary
gregariously
hyperanxious
hypercautious
hyperfastidious
inconsequentially
instantaneously
intravenously
mendaciously
miscellaneously
nefariously
neurologically
neurotically
ostentatiously
outwearying
postrevolutionary
precariously
precautionary
prerevolutionary
revolutionary
sacrilegiously
simultaneously
tenaciously
uncomplimentary
unconventionally
unequivocally
unintentionally
unquestionably


#### Exercise 6  

*Write a function called `is_abecedarian` that returns `True` if the letters in a word appear in alphabetical order (double letters are ok). How many abecedarian words are there?*

In [32]:
def is_abecedarian(word):
    i = 0
    while i < len(word) - 1:
        if word[i] <= word[i + 1]:
            i += 1
        else:
            return False
    return True


In [33]:
is_abecedarian("about")

False

In [34]:
is_abecedarian("abbot")

True

*__Technically, looping with indices isn't introduced until the next section of this chapter; but the alternatives (which are presented in 9.4) seem clunky in comparison.__*

In [35]:
# find abecedarian words:

fin = open('words.txt')

abc_tally = 0

for line in fin:
    word = line.strip()
    if is_abecedarian(word):
        print(word)
        abc_tally += 1
        
print("\nThere are", abc_tally, "abecedarian words in the list.")

aa
aah
aahs
aal
aals
aas
abbe
abbes
abbess
abbey
abbot
abet
abhor
abhors
ably
abo
abort
abos
abuzz
aby
accent
accept
access
accost
ace
acers
aces
achoo
achy
act
ad
add
adder
adders
adds
adeem
adeems
adept
adios
adit
ado
adopt
ados
ads
adz
ae
aegis
aery
aff
affix
afflux
afoot
aft
agin
agio
agios
agist
aglow
agly
ago
ah
ahoy
ai
ail
ails
aim
aims
ain
ains
air
airs
airt
airy
ais
ait
all
allot
allow
alloy
alls
ally
almost
alms
alow
alp
alps
alt
am
ammo
ammos
amort
amp
amps
amu
an
annoy
ant
any
apt
ar
ars
art
arty
as
ass
at
aw
ay
be
bee
beef
beefily
beefs
beefy
been
beep
beeps
beer
beers
beery
bees
beet
befit
beg
begin
begins
begirt
begot
begs
beknot
bel
bell
bellow
bells
belly
below
bels
belt
ben
benny
bens
bent
berry
best
bet
bevy
bey
bhoot
bi
bijou
bijoux
bill
billow
billowy
bills
billy
bin
bins
bint
bio
biopsy
bios
birr
birrs
bis
bit
bitt
bitty
bloop
bloops
blot
blotty
blow
blowy
bo
boo
boor
boors
boos
boost
boot
booty
bop
bops
bort
borty
bortz
bos
boss
bossy
bot
bott
bow
box
boxy
boy
bu

### 9.3 Search

__Reduction to a previously solved problem:__ `uses_all` could have been written using `uses_only`:



In [36]:
def uses_all(word, required):
    return uses_only(required, word)

In [37]:
# Words that use all vowels and 'y' at least once:

fin = open('words.txt')
for line in fin:
    word = line.strip()
    if uses_all(word, 'aeiouy'):
        print(word)

abstemiously
adventitiously
aeronautically
ambidextrously
antievolutionary
antirevolutionary
antiunemployment
authoritatively
autotypies
buoyancies
counterinflationary
evolutionary
extracommunity
facetiously
genitourinary
gregariously
hyperanxious
hypercautious
hyperfastidious
inconsequentially
instantaneously
intravenously
mendaciously
miscellaneously
nefariously
neurologically
neurotically
ostentatiously
outwearying
postrevolutionary
precariously
precautionary
prerevolutionary
revolutionary
sacrilegiously
simultaneously
tenaciously
uncomplimentary
unconventionally
unequivocally
unintentionally
unquestionably


### 9.4 Looping with indices

*__No notes.__*

### 9.5 Debugging

*__No notes.__*

### 9.6 Glossary

*__No notes.__*

### 9.7 Exercises

#### Exercise 7  

*This question is based on a Puzzler that was broadcast on the radio program Car Talk (http://www.cartalk.com/content/puzzlers):*

> *Give me a word with three consecutive double letters. I’ll give you a couple of words that almost qualify, but don’t. For example, the word committee, c-o-m-m-i-t-t-e-e. It would be great except for the ‘i’ that sneaks in there. Or Mississippi: M-i-s-s-i-s-s-i-p-p-i. If you could take out those i’s it would work. But there is a word that has three consecutive pairs of letters and to the best of my knowledge this may be the only word. Of course there are probably 500 more but I can only think of one. What is the word?*

*Write a program to find it.*

In [38]:
def has_three_consecutive_double_letters(w):
    i = 0
    while i < (len(w) - 5):
        if w[i] == w[i + 1] and w[i + 2] == w[i + 3] and w[i + 4] == w[i + 5]:
            return True
        i += 1
    return False
        
        

In [39]:
has_three_consecutive_double_letters('sesquipedalian')

False

In [40]:
has_three_consecutive_double_letters('abcdeffgghh')

True

In [41]:
# Words with three consecutive double letters:

fin = open('words.txt')
for line in fin:
    word = line.strip()
    if has_three_consecutive_double_letters(word):
        print(word)

bookkeeper
bookkeepers
bookkeeping
bookkeepings


#### Exercise 8   Here’s another Car Talk Puzzler (http://www.cartalk.com/content/puzzlers):

> *“I was driving on the highway the other day and I happened to notice my odometer. Like most odometers, it shows six digits, in whole miles only. So, if my car had 300,000 miles, for example, I’d see 3-0-0-0-0-0.
“Now, what I saw that day was very interesting. I noticed that the last 4 digits were palindromic; that is, they read the same forward as backward. For example, 5-4-4-5 is a palindrome, so my odometer could have read 3-1-5-4-4-5.*

> *“One mile later, the last 5 numbers were palindromic. For example, it could have read 3-6-5-4-5-6. One mile after that, the middle 4 out of 6 numbers were palindromic. And you ready for this? One mile later, all 6 were palindromic!*

> *“The question is, what was on the odometer when I first looked?”*

*Write a Python program that tests all the six-digit numbers and prints any numbers that satisfy these requirements.* 

In [42]:
def is_palindrome(word):
    """
    Returns True if word is palindromic.
    """
    return word == word[::-1]

In [43]:
for i in range(100000, 999999):
    if is_palindrome(str(i)[2:]):
        if is_palindrome(str(i + 1)[1:]):
            if is_palindrome(str(i + 2)[1:5]):
                if is_palindrome(str(i + 3)):
                    print(i)

198888
199999


*__While both 198888 and 199999 satisfy the requirements of the puzzle, it seems that 198888 is the better answer.  In 198888 the last four digits are palindromic; in 198889 the last five are; in 198890 the middle four are; and in 198891 all six are.  However, in 199999, the last five - not four - digits - are palindromic, and it seems the creator of the puzzle would have specified this if that is what he was looking for.__*

#### Exercise 9   Here’s another Car Talk Puzzler you can solve with a search (http://www.cartalk.com/content/puzzlers):

> *“Recently I had a visit with my mom and we realized that the two digits that make up my age when reversed resulted in her age. For example, if she’s 73, I’m 37. We wondered how often this has happened over the years but we got sidetracked with other topics and we never came up with an answer.
“When I got home I figured out that the digits of our ages have been reversible six times so far. I also figured out that if we’re lucky it would happen again in a few years, and if we’re really lucky it would happen one more time after that. In other words, it would have happened 8 times over all. So the question is, how old am I now?”*

*Write a Python program that searches for solutions to this Puzzler. Hint: you might find the string method `zfill` useful.*

In [44]:
def are_palindromes(word1, word2):
    """
    Returns True if word1 and word2 are palindromic.
    """
    return word1 == word2[::-1]

# Initialize counts at zero

num_reversed_ages = 0
age_difference_with_most_reversed_ages = 0

# Search for occurrences of reversed ages for a reasonable range
# of age differences, e.g. 10 - 55 (i.e., it's crazy to think 
# his mother was younger than 9 or older than 56 when she had him).

for age_difference in range(10, 55):
    reversed_ages = 0
    
    # tally the number of reversed ages for this age difference
    for my_age in range(0, 99):
        moms_age = my_age + age_difference
        
        # this riddle won't work for triple-digit ages
        if moms_age < 100:
            if are_palindromes(str(my_age).zfill(2), str(moms_age)):
                reversed_ages += 1
    
    # reset counts if we have a new maximum
    if reversed_ages > num_reversed_ages:
        num_reversed_ages = reversed_ages
        age_difference_with_most_reversed_ages = age_difference
        reversed_ages = 0
        
print("The age difference with the most reversed ages:",
      age_difference_with_most_reversed_ages)
print("Number of reversed ages:", num_reversed_ages)

The age difference with the most reversed ages: 18
Number of reversed ages: 8


In [45]:
# print out reversed ages for this difference

for my_age in range(0, 99):
    age_difference = 18
    moms_age = my_age + age_difference
    if moms_age < 100:
        if are_palindromes(str(my_age).zfill(2), str(moms_age)):
            print(str(my_age).zfill(2), " ", str(moms_age))

02   20
13   31
24   42
35   53
46   64
57   75
68   86
79   97
